{"metadata":{"kernelspec":{"name":"ir","display_name":"R","language":"R"},"language_info":{"name":"R","codemirror_mode":"r","pygments_lexer":"r","mimetype":"text/x-r-source","file_extension":".r","version":"4.4.0"},"kaggle":{"accelerator":"none","dataSources":[{"sourceId":12129893,"sourceType":"datasetVersion","datasetId":7638376}],"isInternetEnabled":true,"language":"r","sourceType":"notebook","isGpuEnabled":false}},"nbformat_minor":4,"nbformat":4,"cells":[{"cell_type":"markdown","source":"# An Analysis of Cell Types found in the Primary Motor Cortex (M1) of *Homo sapiens*\n\nThis project attempts to correlate the biological functions of different M1 cell types with their phenotypic expression.\n\nThe project uses **single-nucleus RNA-seq** data, obtained from the study [Bakken et al., 2021](https://www.nature.com/articles/s41586-021-03465-8#Abs1). It is available to download from the [Allen Brain Institute](https://portal.brain-map.org/atlases-and-data/rnaseq/human-m1-10x).","metadata":{}},{"cell_type":"markdown","source":"## 1. Install and load the required packages.","metadata":{}},{"cell_type":"code","source":"if (!require(\"BiocManager\", quietly = TRUE))\n    install.packages(\"BiocManager\", version = \"3.21\")","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-11T11:31:30.801908Z","iopub.execute_input":"2025-06-11T11:31:30.803562Z","iopub.status.idle":"2025-06-11T11:31:30.815799Z","shell.execute_reply":"2025-06-11T11:31:30.813981Z"}},"outputs":[],"execution_count":9},{"cell_type":"code","source":"BiocManager::install(c(\"DESeq2\", \"biomaRt\"), quietly = TRUE)\ninstall.packages(c(\"tidyverse\", \"pheatmap\"), quietly = TRUE)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-21T07:23:22.191384Z","iopub.execute_input":"2025-06-21T07:23:22.193067Z","iopub.status.idle":"2025-06-21T07:33:28.700304Z","shell.execute_reply":"2025-06-21T07:33:28.698252Z"}},"outputs":[{"name":"stderr","text":"'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n'help(\"repositories\", package = \"BiocManager\")' for details.\nReplacement repositories:\n    CRAN: http://cran.rstudio.com/\n\nBioconductor version 3.19 (BiocManager 1.30.23), R 4.4.0 (2024-04-24)\n\nInstalling package(s) 'DESeq2', 'biomaRt'\n\nalso installing the dependencies ‘UCSC.utils’, ‘GenomeInfoDbData’, ‘zlibbioc’, ‘SparseArray’, ‘Biostrings’, ‘GenomeInfoDb’, ‘XVector’, ‘S4Arrays’, ‘DelayedArray’, ‘KEGGREST’, ‘S4Vectors’, ‘IRanges’, ‘GenomicRanges’, ‘SummarizedExperiment’, ‘Biobase’, ‘BiocParallel’, ‘MatrixGenerics’, ‘AnnotationDbi’, ‘BiocFileCache’\n\n\nOld packages: 'abind', 'actuar', 'ade4', 'adegraphics', 'adehabitatHR',\n  'adehabitatLT', 'adehabitatMA', 'admisc', 'AER', 'afex', 'aglm', 'AID',\n  'akc', 'ale', 'AlgDesign', 'AlphaSimR', 'amap', 'Amelia', 'anacor',\n  'antiword', 'anytime', 'ape', 'aplot', 'apollo', 'aqp', 'arrow', 'ars',\n  'ARTool', 'AsioHeaders', 'askpass', 'ASMap', 'astsa', 'automap', 'baguette',\n  'BAS', 'base64', 'bayesplot', 'bayestestR', 'bbotk', 'BDgraph', 'bedr',\n  'beepr', 'Benchmarking', 'betareg', 'BH', 'bibliometrix', 'bigD',\n  'bigstatsr', 'bigstep', 'billboarder', 'bindr', 'BiocManager',\n  'BiodiversityR', 'biomod2', 'biotools', 'bipartite', 'bit', 'bit64',\n  'bitops', 'blme', 'blorr', 'BMA', 'BMisc', 'bnlearn', 'bonsai', 'bookdown',\n  'boot', 'brms', 'brnn', 'broom', 'broom.helpers', 'broom.mixed', 'brulee',\n  'bslib', 'bundle', 'butcher', 'C50', 'calculus', 'car', 'caret',\n  'caretEnsemble', 'caTools', 'cba', 'censored', 'changepoint', 'cheapr',\n  'checkmate', 'chk', 'CholWishart', 'choroplethr', 'chromote', 'chron',\n  'CircStats', 'circular', 'class', 'classInt', 'cld2', 'cleaner', 'cli',\n  'climatol', 'clock', 'clue', 'cluster', 'clusterSim', 'ClustOfVar',\n  'cmsafops', 'coarseDataTools', 'cobalt', 'cobs', 'collapse', 'collections',\n  'colorspace', 'commonmark', 'compareGroups', 'Compind', 'condMVNorm',\n  'conjoint', 'ConsRank', 'contentid', 'copula', 'CORElearn', 'coro',\n  'correlation', 'corrplot', 'COUNT', 'countrycode', 'cpp11', 'cppdoubles',\n  'credentials', 'cricketdata', 'crul', 'crypto2', 'cubature', 'Cubist',\n  'curl', 'cutpointr', 'cvms', 'data.table', 'dataMaid', 'datamods',\n  'datasauRus', 'datawizard', 'datetimeutils', 'dbarts', 'dbscan',\n  'dendextend', 'dendroTools', 'DEoptimR', 'Deriv', 'descriptio', 'DescTools',\n  'devEMF', 'dfidx', 'dgof', 'DHARMa', 'dials', 'diffobj', 'digest', 'dimRed',\n  'directlabels', 'dismo', 'distances', 'distillery', 'distr', 'distrEx',\n  'distributional', 'DistributionUtils', 'dlm', 'dlnm', 'dm', 'doBy', 'docopt',\n  'DoE.base', 'doFuture', 'doRNG', 'DoseFinding', 'dotCall64', 'dotwhisker',\n  'DoubleML', 'downloader', 'dplR', 'drat', 'DRDID', 'dreamerr', 'dtt',\n  'dtwclust', 'duckdb', 'e1071', 'earth', 'Ecfun', 'ecospat', 'ecp',\n  'effectsize', 'eha', 'embed', 'emmeans', 'energy', 'entropy', 'EnvStats',\n  'Epi', 'epiR', 'erer', 'ergm', 'ergm.multi', 'esquisse', 'estimatr', 'etm',\n  'evaluate', 'evd', 'Exact', 'explore', 'expm', 'ExPosition', 'extRemes',\n  'fable', 'fabletools', 'faraway', 'fastcluster', 'fastDummies', 'fastICA',\n  'fastmatch', 'fastmatrix', 'fastrmodels', 'fastverse', 'fBasics', 'fda',\n  'fda.usc', 'fdapace', 'fdrtool', 'feasts', 'ff', 'fftw', 'FielDHub',\n  'fields', 'filehashSQLite', 'finalfit', 'finetune', 'fitdistrplus', 'FKF',\n  'flare', 'flexclust', 'flexmix', 'flexsurv', 'flextable', 'float', 'FNN',\n  'fontawesome', 'forecast', 'foreign', 'forestplot', 'forestploter', 'fpc',\n  'fpp3', 'FrF2', 'fs', 'FSA', 'FSelectorRcpp', 'fstcore', 'ftsa', 'future',\n  'future.apply', 'gam', 'gamboostLSS', 'gamm4', 'gap', 'gapminder', 'garma',\n  'GAS', 'gclus', 'gcookbook', 'gdata', 'GDAtools', 'gdtools', 'geepack',\n  'GeneCycle', 'GeneralizedHyperbolic', 'generics', 'genridge', 'GenSA',\n  'geobr', 'geodist', 'geojsonR', 'geometry', 'geomorph', 'geomtextpath',\n  'geoR', 'geosphere', 'gert', 'GetBCBData', 'ggdag', 'ggdist', 'ggeasy',\n  'ggeffects', 'ggforce', 'ggfun', 'ggh4x', 'gghighlight', 'ggiraph',\n  'gglasso', 'ggmap', 'GGMridge', 'ggnewscale', 'ggpath', 'ggpattern',\n  'ggplate', 'ggplot2', 'ggpmisc', 'ggprism', 'ggrepel', 'ggstats', 'gh',\n  'ghyp', 'gifski', 'giscoR', 'git2r', 'glarma', 'gld', 'GLMMadaptive',\n  'glmmTMB', 'glmnet', 'glmtoolbox', 'globals', 'glue', 'gmapsdistance', 'gmp',\n  'GoFKernel', 'golem', 'googlePolylines', 'gower', 'GPArotation', 'GPfit',\n  'gplots', 'grafify', 'gRain', 'graphlayouts', 'gRbase', 'greybox', 'grf',\n  'gridpattern', 'grImport2', 'groupdata2', 'grpreg', 'gsheet', 'gss', 'gstat',\n  'gsw', 'gt', 'gtable', 'gtExtras', 'gtsummary', 'gutenbergr', 'GWalkR',\n  'GWmodel', 'h3lib', 'h3r', 'hardhat', 'harmony', 'haven', 'hdf5r',\n  'HelpersMG', 'heplots', 'hexbin', 'HistData', 'HMDHFDplus', 'Hmisc', 'HSAUR',\n  'htmlTable', 'hts', 'httpuv', 'httr2', 'hunspell', 'huxtable', 'hydroGOF',\n  'hydroTSM', 'hypergeo', 'igraph', 'ijtiff', 'imager', 'iml', 'incidence',\n  'infer', 'inline', 'insight', 'inspectdf', 'interactions', 'intervals',\n  'ipred', 'ISOcodes', 'isotree', 'ISwR', 'itsmr', 'ivreg', 'janitor',\n  'JMbayes2', 'jmvconnect', 'jmvcore', 'jmvReadWrite', 'jpeg', 'jqr',\n  'jsonlite', 'jtools', 'kdensity', 'keras3', 'kernelshap', 'kernlab',\n  'KernSmooth', 'keyring', 'KFAS', 'khroma', 'kit', 'kknn', 'KMsurv', 'knitr',\n  'ks', 'kyotil', 'L1pack', 'labelled', 'LaF', 'Lahman', 'languageR', 'lares',\n  'later', 'lattice', 'LatticeKrig', 'lava', 'lavaan', 'lazyWeave', 'lcmm',\n  'leafem', 'leafgl', 'lehdr', 'lemon', 'lessR', 'lfe', 'LiblineaR',\n  'lightgbm', 'limma', 'linelist', 'lintr', 'littler', 'lme4', 'lmodel2',\n  'lmom', 'locfit', 'logger', 'logistf', 'longitudinalData', 'loo', 'lordif',\n  'LPCM', 'lpSolve', 'lpSolveAPI', 'lsmeans', 'ltsa', 'lubridate',\n  'MachineShop', 'maditr', 'magick', 'manymome', 'maplegend', 'mapproj',\n  'maps', 'mapsf', 'marginaleffects', 'margins', 'markdown', 'markovchain',\n  'MASS', 'Matching', 'MatchIt', 'mathjaxr', 'matlib', 'Matrix',\n  'MatrixModels', 'matrixStats', 'maxstat', 'MazamaCoreUtils',\n  'MazamaSpatialUtils', 'MBA', 'mboost', 'MCMCpack', 'mda', 'mdsr',\n  'mediation', 'memisc', 'meta', 'metacoder', 'metadat', 'metafor',\n  'meteospain', 'MetricsWeighted', 'mets', 'mfx', 'mgcv', 'mgm', 'mgwrsar',\n  'mice', 'microbenchmark', 'midasr', 'mime', 'miniUI', 'minqa', 'minty',\n  'mirt', 'mirtCAT', 'missRanger', 'misty', 'mixlm', 'mixtools', 'mlbench',\n  'mlflow', 'mlogit', 'mlr3', 'mlr3benchmark', 'mlr3cluster', 'mlr3data',\n  'mlr3filters', 'mlr3fselect', 'mlr3learners', 'mlr3mbo', 'mlr3measures',\n  'mlr3misc', 'mlr3pipelines', 'mlr3tuning', 'mlr3tuningspaces', 'mlr3verse',\n  'mlr3viz', 'modelbased', 'modelenv', 'modelsummary', 'modeltime',\n  'modeltools', 'moderndive', 'mongolite', 'monitoR', 'msigdbr', 'msm', 'msme',\n  'mstate', 'mtsdi', 'multcomp', 'MuMIn', 'MVN', 'mvnormalTest', 'mvtnorm',\n  'NatureSounds', 'ncdf4', 'ncdump', 'ncvreg', 'ndjson', 'ndtv', 'network',\n  'networkD3', 'networkDynamic', 'networkLite', 'nFactors', 'nflfastR',\n  'nflplotR', 'nflreadr', 'nflseedR', 'nhanesA', 'nimble', 'nlme', 'nloptr',\n  'NLP', 'NMOF', 'nnet', 'nnls', 'NNS', 'nomclust', 'np', 'nullabor', 'oce',\n  'odbc', 'officer', 'olsrr', 'oompaBase', 'openalexR', 'opencv',\n  'opendatatoronto', 'openeo', 'openssl', 'openxlsx', 'optimx', 'ordinal',\n  'ordinalForest', 'ordinalNet', 'ore', 'osmextract', 'overlapping',\n  'packcircles', 'packHV', 'packrat', 'padr', 'pagedown', 'pammtools',\n  'pander', 'parabar', 'paradox', 'parallelly', 'parameters', 'ParamHelpers',\n  'paran', 'parsnip', 'partitions', 'party', 'partykit', 'patchwork', 'pbdZMQ',\n  'pbkrtest', 'pcaPP', 'pdftools', 'pdp', 'pdR', 'PearsonDS', 'performance',\n  'PerformanceAnalytics', 'pgmm', 'phangorn', 'pheatmap', 'phia',\n  'philentropy', 'phytools', 'pillar', 'pingr', 'pins', 'pivottabler',\n  'pkgbuild', 'pkgdown', 'pliman', 'plm', 'plotly', 'plotmo', 'PlotTools',\n  'pls', 'plsVarSel', 'plumber', 'PMA', 'PMwR', 'poibin', 'polyclip',\n  'popdemo', 'portfolio', 'PortfolioAnalytics', 'posterior', 'POT', 'poweRlaw',\n  'prabclus', 'precrec', 'PreProcess', 'prettyGraphs', 'priceR', 'probably',\n  'processx', 'prodlim', 'profvis', 'progressr', 'PROJ', 'proj4', 'promises',\n  'prospectr', 'protolite', 'proxyC', 'PRROC', 'ps', 'pspline', 'PSweight',\n  'psych', 'purrr', 'qdapRegex', 'qgam', 'qpcR', 'qpdf', 'qs', 'qtl',\n  'qtlcharts', 'quanteda', 'quantmod', 'quantreg', 'quarto', 'questionr',\n  'QuickJSR', 'qvcalc', 'R.cache', 'R.oo', 'R.utils', 'r2rtf', 'R6',\n  'radiant.data', 'radiant.model', 'radiant.multivariate', 'ragg', 'rAmCharts',\n  'randomForest', 'randomForestSRC', 'randtoolbox', 'ranger', 'RANN',\n  'RApiSerialize', 'rapportools', 'raster', 'rayimage', 'rayrender',\n  'rayvertex', 'rbibutils', 'Rborist', 'Rcmdr', 'Rcompadre', 'rcompanion',\n  'Rcpp', 'RcppArmadillo', 'RcppDE', 'RcppEigen', 'RcppParallel', 'RcppRoll',\n  'RcppSimdJson', 'RcppThread', 'RcppTOML', 'RcppZiggurat', 'RCurl', 'Rdpack',\n  'Rdsdp', 'reactR', 'readODS', 'readstata13', 'readxl', 'rearrr', 'rebmix',\n  'recipes', 'refund', 'regclass', 'rentrez', 'renv', 'report', 'reprex',\n  'reshape', 'reticulate', 'Rfast', 'rfishbase', 'rgbif', 'rgenoud', 'rgl',\n  'rio', 'riskRegression', 'rjson', 'RJSONIO', 'rlang', 'rle', 'RMariaDB',\n  'rmarkdown', 'rminer', 'rms', 'RMySQL', 'rngWELL', 'RNifti', 'RNiftyReg',\n  'robotstxt', 'robust', 'robustbase', 'RODBC', 'rpart', 'rpf', 'rpostgis',\n  'RPostgres', 'RPostgreSQL', 'rrcov', 'rrcovHD', 'rredlist', 'RRPP', 'RSAGA',\n  'rsample', 'rsconnect', 'rscopus', 'Rserve', 'rsm', 'Rsolnp', 'rsparse',\n  'RSpectra', 'RSQLite', 'Rssa', 'rstac', 'rstan', 'rstpm2', 'rstudioapi',\n  'rsvg', 'rtkore', 'rugarch', 'RUnit', 'Rvcg', 'rvg', 's2', 'sandwich',\n  'sass', 'scales', 'SciViews', 'SCtools', 'sctransform', 'sda', 'sdm', 'seas',\n  'seasonal', 'seasonalview', 'see', 'seededlda', 'segmented', 'sem',\n  'semTools', 'sensemakr', 'sensitivity', 'seriation', 'servr', 'sessioninfo',\n  'Seurat', 'SeuratObject', 'sf', 'sfnetworks', 'sfsmisc', 'sftime',\n  'shadowtext', 'shapr', 'shapviz', 'shiny', 'shinyAce', 'shinycssloaders',\n  'shinydashboard', 'shinyWidgets', 'SimDesign', 'simmer', 'simpleboot',\n  'simputation', 'sits', 'sjPlot', 'sjstats', 'skmeans', 'slam', 'slider',\n  'SLOPE', 'smaa', 'smacof', 'smooth', 'smplot2', 'sna', 'sodium',\n  'softImpute', 'soilDB', 'sommer', 'sp', 'spacefillr', 'spacetime', 'spam',\n  'sparklyr', 'sparseFLMM', 'SparseM', 'spatialreg', 'spatialrisk', 'spatstat',\n  'spatstat.data', 'spatstat.explore', 'spatstat.geom', 'spatstat.linnet',\n  'spatstat.model', 'spatstat.random', 'spatstat.utils', 'spData', 'spdep',\n  'spmodel', 'spmoran', 'spotifyr', 'SSN2', 'stabledist', 'StableEstim',\n  'stacks', 'StanHeaders', 'stars', 'startupmsg', 'statgenGWAS',\n  'statnet.common', 'stdmod', 'steadyICA', 'StepReg', 'strex', 'stringdist',\n  'stringi', 'stringmagic', 'strucchange', 'summarytools', 'SuppDists',\n  'survival', 'survminer', 'svd', 'svglite', 'svgtools', 'sys', 'systemfonts',\n  'tables', 'tabnet', 'tau', 'taxize', 'tcltk2', 'TDboost', 'tensor', 'tergm',\n  'Ternary', 'terra', 'tesseract', 'testthat', 'texreg', 'text', 'textcat',\n  'textrecipes', 'textshaping', 'textutils', 'tfdatasets', 'tgp', 'TH.data',\n  'thematic', 'themis', 'threejs', 'tibble', 'tibbletime', 'tidybayes',\n  'tidycensus', 'tidycmprsk', 'tidyfst', 'tidygeocoder', 'tidymodels',\n  'tidypredict', 'tidyquant', 'tidytable', 'tidytuesdayR', 'tigris',\n  'timeDate', 'timeplyr', 'timereg', 'timeSeries', 'tinytable', 'tinytex',\n  'tm', 'tmap', 'tmaptools', 'TMB', 'toastui', 'topicmodels', 'torch', 'tourr',\n  'TraMineR', 'tree', 'TruncatedNormal', 'tsbox', 'TSclust', 'tsDyn',\n  'tseries', 'tsibble', 'tsna', 'TSP', 'tswge', 'tuber', 'tune', 'tzdb',\n  'Unicode', 'units', 'univariateML', 'unpivotr', 'urltools', 'usethis',\n  'usmap', 'usmapdata', 'utf8', 'uuid', 'uwot', 'V8', 'vcd', 'vegan',\n  'verification', 'VGAM', 'vioplot', 'vtable', 'waldo', 'warbleR',\n  'WaveletComp', 'wbacon', 'WDI', 'webshot2', 'websocket', 'webutils',\n  'weights', 'WeightSVM', 'winch', 'withr', 'wk', 'wooldridge', 'workflows',\n  'workflowsets', 'writexl', 'WriteXLS', 'xfun', 'xgboost', 'XML', 'xml2',\n  'xplorerr', 'xts', 'yahoofinancer', 'yaImpute', 'yaml', 'yardstick', 'yfR',\n  'yuima', 'yulab.utils', 'zeallot', 'zip', 'zoo', 'spatial'\n\nInstalling packages into ‘/usr/local/lib/R/site-library’\n(as ‘lib’ is unspecified)\n\n","output_type":"stream"}],"execution_count":3},{"cell_type":"code","source":"library(DESeq2)\nlibrary(biomaRt)\nlibrary(tidyverse)\nlibrary(pheatmap)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-21T07:33:28.703291Z","iopub.execute_input":"2025-06-21T07:33:28.704808Z","iopub.status.idle":"2025-06-21T07:33:39.422995Z","shell.execute_reply":"2025-06-21T07:33:39.421005Z"}},"outputs":[{"name":"stderr","text":"Loading required package: S4Vectors\n\nLoading required package: stats4\n\nLoading required package: BiocGenerics\n\n\nAttaching package: ‘BiocGenerics’\n\n\nThe following objects are masked from ‘package:stats’:\n\n    IQR, mad, sd, var, xtabs\n\n\nThe following objects are masked from ‘package:base’:\n\n    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,\n    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,\n    Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,\n    tapply, union, unique, unsplit, which.max, which.min\n\n\n\nAttaching package: ‘S4Vectors’\n\n\nThe following object is masked from ‘package:utils’:\n\n    findMatches\n\n\nThe following objects are masked from ‘package:base’:\n\n    expand.grid, I, unname\n\n\nLoading required package: IRanges\n\nLoading required package: GenomicRanges\n\nLoading required package: GenomeInfoDb\n\nLoading required package: SummarizedExperiment\n\nLoading required package: MatrixGenerics\n\nLoading required package: matrixStats\n\n\nAttaching package: ‘MatrixGenerics’\n\n\nThe following objects are masked from ‘package:matrixStats’:\n\n    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n    colWeightedMeans, colWeightedMedians, colWeightedSds,\n    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n    rowWeightedSds, rowWeightedVars\n\n\nLoading required package: Biobase\n\nWelcome to Bioconductor\n\n    Vignettes contain introductory material; view with\n    'browseVignettes()'. To cite Bioconductor, see\n    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n\n\n\nAttaching package: ‘Biobase’\n\n\nThe following object is masked from ‘package:MatrixGenerics’:\n\n    rowMedians\n\n\nThe following objects are masked from ‘package:matrixStats’:\n\n    anyMissing, rowMedians\n\n\nThe following object is masked from ‘package:httr’:\n\n    content\n\n\n── \u001b[1mAttaching core tidyverse packages\u001b[22m ──────────────────────── tidyverse 2.0.0 ──\n\u001b[32m✔\u001b[39m \u001b[34mdplyr    \u001b[39m 1.1.4     \u001b[32m✔\u001b[39m \u001b[34mreadr    \u001b[39m 2.1.5\n\u001b[32m✔\u001b[39m \u001b[34mforcats  \u001b[39m 1.0.0     \u001b[32m✔\u001b[39m \u001b[34mstringr  \u001b[39m 1.5.1\n\u001b[32m✔\u001b[39m \u001b[34mggplot2  \u001b[39m 3.5.1     \u001b[32m✔\u001b[39m \u001b[34mtibble   \u001b[39m 3.2.1\n\u001b[32m✔\u001b[39m \u001b[34mlubridate\u001b[39m 1.9.3     \u001b[32m✔\u001b[39m \u001b[34mtidyr    \u001b[39m 1.3.1\n\u001b[32m✔\u001b[39m \u001b[34mpurrr    \u001b[39m 1.0.2     \n── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32m%within%()\u001b[39m masks \u001b[34mIRanges\u001b[39m::%within%()\n\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcollapse()\u001b[39m     masks \u001b[34mIRanges\u001b[39m::collapse()\n\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcombine()\u001b[39m      masks \u001b[34mBiobase\u001b[39m::combine(), \u001b[34mBiocGenerics\u001b[39m::combine()\n\u001b[31m✖\u001b[39m \u001b[34mBiobase\u001b[39m::\u001b[32mcontent()\u001b[39m    masks \u001b[34mhttr\u001b[39m::content()\n\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcount()\u001b[39m        masks \u001b[34mmatrixStats\u001b[39m::count()\n\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mdesc()\u001b[39m         masks \u001b[34mIRanges\u001b[39m::desc()\n\u001b[31m✖\u001b[39m \u001b[34mtidyr\u001b[39m::\u001b[32mexpand()\u001b[39m       masks \u001b[34mS4Vectors\u001b[39m::expand()\n\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m       masks \u001b[34mstats\u001b[39m::filter()\n\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfirst()\u001b[39m        masks \u001b[34mS4Vectors\u001b[39m::first()\n\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m          masks \u001b[34mstats\u001b[39m::lag()\n\u001b[31m✖\u001b[39m \u001b[34mggplot2\u001b[39m::\u001b[32mPosition()\u001b[39m   masks \u001b[34mBiocGenerics\u001b[39m::Position(), \u001b[34mbase\u001b[39m::Position()\n\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mreduce()\u001b[39m       masks \u001b[34mGenomicRanges\u001b[39m::reduce(), \u001b[34mIRanges\u001b[39m::reduce()\n\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mrename()\u001b[39m       masks \u001b[34mS4Vectors\u001b[39m::rename()\n\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32msecond()\u001b[39m   masks \u001b[34mS4Vectors\u001b[39m::second()\n\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32msecond<-()\u001b[39m masks \u001b[34mS4Vectors\u001b[39m::second<-()\n\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mselect()\u001b[39m       masks \u001b[34mbiomaRt\u001b[39m::select()\n\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mslice()\u001b[39m        masks \u001b[34mIRanges\u001b[39m::slice()\n\u001b[36mℹ\u001b[39m Use the conflicted package (\u001b[3m\u001b[34m<http://conflicted.r-lib.org/>\u001b[39m\u001b[23m) to force all conflicts to become errors\n","output_type":"stream"}],"execution_count":4},{"cell_type":"markdown","source":"## 2. Read in the expression matrix and the metadata file.","metadata":{}},{"cell_type":"code","source":"options(timeout = 1000)\nmatrix_link <- \"https://idk-etl-prod-download-bucket.s3.amazonaws.com/aibs_human_m1_10x/matrix.csv\"\ngene_matrix <- data.table::fread(matrix_link)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-21T07:35:30.000639Z","iopub.execute_input":"2025-06-21T07:35:30.003391Z","iopub.status.idle":"2025-06-21T07:39:26.952452Z","shell.execute_reply":"2025-06-21T07:39:26.948734Z"}},"outputs":[],"execution_count":5},{"cell_type":"code","source":"meta_filepath <- \"/kaggle/input/metadata/metadata.csv\"\nmeta_data <- read.csv(meta_filepath)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-21T07:41:17.000998Z","iopub.execute_input":"2025-06-21T07:41:17.003364Z","iopub.status.idle":"2025-06-21T07:41:18.465009Z","shell.execute_reply":"2025-06-21T07:41:18.462778Z"}},"outputs":[],"execution_count":6},{"cell_type":"markdown","source":"## 3. Explore and clean the metadata file.","metadata":{}},{"cell_type":"code","source":"str(meta_data)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-17T11:49:42.047472Z","iopub.execute_input":"2025-06-17T11:49:42.049586Z","iopub.status.idle":"2025-06-17T11:49:42.093436Z","shell.execute_reply":"2025-06-17T11:49:42.091379Z"}},"outputs":[{"name":"stdout","text":"'data.frame':\t76533 obs. of  39 variables:\n $ sample_name                : chr  \"AAACCCAAGGATTTCC-LKTX_190129_01_A01\" \"AAACCCAAGTATGGCG-LKTX_190129_01_A01\" \"AAACCCACAAAGTGTA-LKTX_190129_01_A01\" \"AAACCCACACTACTTT-LKTX_190129_01_A01\" ...\n $ exp_component_name         : chr  \"AAACCCAAGGATTTCC-21L8TX_180927_001_A01\" \"AAACCCAAGTATGGCG-21L8TX_180927_001_A01\" \"AAACCCACAAAGTGTA-21L8TX_180927_001_A01\" \"AAACCCACACTACTTT-21L8TX_180927_001_A01\" ...\n $ cluster_label              : chr  \"Inh L1-2 SST CCNJL\" \"Exc L5-6 FEZF2 IFNG-AS1\" \"Exc L3-5 RORB LINC01202\" \"Exc L2 LINC00507 GLRA3\" ...\n $ cluster_color              : chr  \"#fb8d00\" \"#2c815f\" \"#547d7a\" \"#cecd32\" ...\n $ cluster_order              : int  50 116 87 75 120 101 111 90 90 100 ...\n $ class_label                : chr  \"GABAergic\" \"Glutamatergic\" \"Glutamatergic\" \"Glutamatergic\" ...\n $ class_color                : chr  \"#FF7373\" \"#3DCC3D\" \"#3DCC3D\" \"#3DCC3D\" ...\n $ class_order                : int  1 2 2 2 3 2 2 2 2 2 ...\n $ subclass_label             : chr  \"Sst\" \"L5/6 NP\" \"L5 IT\" \"L2/3 IT\" ...\n $ subclass_color             : chr  \"#FF9900\" \"#3E9E64\" \"#50B2AD\" \"#C4EC04\" ...\n $ subclass_order             : int  5 12 8 7 17 13 11 8 8 13 ...\n $ donor_sex_label            : chr  \"F\" \"F\" \"F\" \"F\" ...\n $ donor_sex_color            : chr  \"#FF7373\" \"#FF7373\" \"#FF7373\" \"#FF7373\" ...\n $ donor_sex_order            : int  1 1 1 1 1 1 1 1 1 1 ...\n $ region_label               : chr  \"M1\" \"M1\" \"M1\" \"M1\" ...\n $ region_color               : chr  \"#FF7373\" \"#FF7373\" \"#FF7373\" \"#FF7373\" ...\n $ region_order               : int  1 1 1 1 1 1 1 1 1 1 ...\n $ cortical_layer_label       : chr  \"all\" \"all\" \"all\" \"all\" ...\n $ cortical_layer_color       : chr  \"#FF7373\" \"#FF7373\" \"#FF7373\" \"#FF7373\" ...\n $ cortical_layer_order       : int  1 1 1 1 1 1 1 1 1 1 ...\n $ cell_type_accession_label  : chr  \"CS1912131050\" \"CS1912131116\" \"CS1912131087\" \"CS1912131075\" ...\n $ cell_type_accession_color  : chr  \"#fb8d00\" \"#2c815f\" \"#547d7a\" \"#cecd32\" ...\n $ cell_type_accession_order  : int  50 116 87 75 120 101 111 90 90 100 ...\n $ cell_type_alias_label      : chr  \"Inh L1-2 SST CCNJL\" \"Exc L5-6 FEZF2 IFNG-AS1\" \"Exc L3-5 RORB LINC01202\" \"Exc L2 LINC00507 GLRA3\" ...\n $ cell_type_alias_color      : chr  \"#fb8d00\" \"#2c815f\" \"#547d7a\" \"#cecd32\" ...\n $ cell_type_alias_order      : int  50 116 87 75 120 101 111 90 90 100 ...\n $ cell_type_alt_alias_label  : chr  \"\" \"\" \"\" \"\" ...\n $ cell_type_alt_alias_color  : chr  \"#fb8d00\" \"#2c815f\" \"#547d7a\" \"#cecd32\" ...\n $ cell_type_alt_alias_order  : int  50 116 87 75 120 101 111 90 90 100 ...\n $ cell_type_designation_label: chr  \"Neuron 50\" \"Neuron 116\" \"Neuron 87\" \"Neuron 75\" ...\n $ cell_type_designation_color: chr  \"#fb8d00\" \"#2c815f\" \"#547d7a\" \"#cecd32\" ...\n $ cell_type_designation_order: int  50 116 87 75 120 101 111 90 90 100 ...\n $ external_donor_name_label  : chr  \"H18.30.001\" \"H18.30.001\" \"H18.30.001\" \"H18.30.001\" ...\n $ external_donor_name_color  : chr  \"#FF7373\" \"#FF7373\" \"#FF7373\" \"#FF7373\" ...\n $ external_donor_name_order  : int  1 1 1 1 1 1 1 1 1 1 ...\n $ specimen_type              : chr  \"nucleus\" \"nucleus\" \"nucleus\" \"nucleus\" ...\n $ full_genotype_label        : logi  NA NA NA NA NA NA ...\n $ outlier_call               : chr  \"False\" \"False\" \"False\" \"False\" ...\n $ outlier_type               : logi  NA NA NA NA NA NA ...\n","output_type":"stream"}],"execution_count":26},{"cell_type":"code","source":"meta_data <- meta_data %>% select(!ends_with(\"_color\") & !ends_with(\"_order\"))\ncolnames(meta_data)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-21T07:41:31.359397Z","iopub.execute_input":"2025-06-21T07:41:31.361391Z","iopub.status.idle":"2025-06-21T07:41:31.403626Z","shell.execute_reply":"2025-06-21T07:41:31.401247Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<style>\n.list-inline {list-style: none; margin:0; padding: 0}\n.list-inline>li {display: inline-block}\n.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n</style>\n<ol class=list-inline><li>'sample_name'</li><li>'exp_component_name'</li><li>'cluster_label'</li><li>'class_label'</li><li>'subclass_label'</li><li>'donor_sex_label'</li><li>'region_label'</li><li>'cortical_layer_label'</li><li>'cell_type_accession_label'</li><li>'cell_type_alias_label'</li><li>'cell_type_alt_alias_label'</li><li>'cell_type_designation_label'</li><li>'external_donor_name_label'</li><li>'specimen_type'</li><li>'full_genotype_label'</li><li>'outlier_call'</li><li>'outlier_type'</li></ol>\n","text/markdown":"1. 'sample_name'\n2. 'exp_component_name'\n3. 'cluster_label'\n4. 'class_label'\n5. 'subclass_label'\n6. 'donor_sex_label'\n7. 'region_label'\n8. 'cortical_layer_label'\n9. 'cell_type_accession_label'\n10. 'cell_type_alias_label'\n11. 'cell_type_alt_alias_label'\n12. 'cell_type_designation_label'\n13. 'external_donor_name_label'\n14. 'specimen_type'\n15. 'full_genotype_label'\n16. 'outlier_call'\n17. 'outlier_type'\n\n\n","text/latex":"\\begin{enumerate*}\n\\item 'sample\\_name'\n\\item 'exp\\_component\\_name'\n\\item 'cluster\\_label'\n\\item 'class\\_label'\n\\item 'subclass\\_label'\n\\item 'donor\\_sex\\_label'\n\\item 'region\\_label'\n\\item 'cortical\\_layer\\_label'\n\\item 'cell\\_type\\_accession\\_label'\n\\item 'cell\\_type\\_alias\\_label'\n\\item 'cell\\_type\\_alt\\_alias\\_label'\n\\item 'cell\\_type\\_designation\\_label'\n\\item 'external\\_donor\\_name\\_label'\n\\item 'specimen\\_type'\n\\item 'full\\_genotype\\_label'\n\\item 'outlier\\_call'\n\\item 'outlier\\_type'\n\\end{enumerate*}\n","text/plain":" [1] \"sample_name\"                 \"exp_component_name\"         \n [3] \"cluster_label\"               \"class_label\"                \n [5] \"subclass_label\"              \"donor_sex_label\"            \n [7] \"region_label\"                \"cortical_layer_label\"       \n [9] \"cell_type_accession_label\"   \"cell_type_alias_label\"      \n[11] \"cell_type_alt_alias_label\"   \"cell_type_designation_label\"\n[13] \"external_donor_name_label\"   \"specimen_type\"              \n[15] \"full_genotype_label\"         \"outlier_call\"               \n[17] \"outlier_type\"               "},"metadata":{}}],"execution_count":7},{"cell_type":"code","source":"head(meta_data)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-18T08:06:57.142968Z","iopub.execute_input":"2025-06-18T08:06:57.145138Z","iopub.status.idle":"2025-06-18T08:06:57.199827Z","shell.execute_reply":"2025-06-18T08:06:57.196431Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<table class=\"dataframe\">\n<caption>A data.frame: 6 × 17</caption>\n<thead>\n\t<tr><th></th><th scope=col>sample_name</th><th scope=col>exp_component_name</th><th scope=col>cluster_label</th><th scope=col>class_label</th><th scope=col>subclass_label</th><th scope=col>donor_sex_label</th><th scope=col>region_label</th><th scope=col>cortical_layer_label</th><th scope=col>cell_type_accession_label</th><th scope=col>cell_type_alias_label</th><th scope=col>cell_type_alt_alias_label</th><th scope=col>cell_type_designation_label</th><th scope=col>external_donor_name_label</th><th scope=col>specimen_type</th><th scope=col>full_genotype_label</th><th scope=col>outlier_call</th><th scope=col>outlier_type</th></tr>\n\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n</thead>\n<tbody>\n\t<tr><th scope=row>1</th><td>AAACCCAAGGATTTCC-LKTX_190129_01_A01</td><td>AAACCCAAGGATTTCC-21L8TX_180927_001_A01</td><td>Inh L1-2 SST CCNJL         </td><td>GABAergic    </td><td>Sst    </td><td>F</td><td>M1</td><td>all</td><td>CS1912131050</td><td>Inh L1-2 SST CCNJL         </td><td>       </td><td>Neuron 50   </td><td>H18.30.001</td><td>nucleus</td><td>NA</td><td>False</td><td>NA</td></tr>\n\t<tr><th scope=row>2</th><td>AAACCCAAGTATGGCG-LKTX_190129_01_A01</td><td>AAACCCAAGTATGGCG-21L8TX_180927_001_A01</td><td>Exc L5-6 FEZF2 IFNG-AS1    </td><td>Glutamatergic</td><td>L5/6 NP</td><td>F</td><td>M1</td><td>all</td><td>CS1912131116</td><td>Exc L5-6 FEZF2 IFNG-AS1    </td><td>       </td><td>Neuron 116  </td><td>H18.30.001</td><td>nucleus</td><td>NA</td><td>False</td><td>NA</td></tr>\n\t<tr><th scope=row>3</th><td>AAACCCACAAAGTGTA-LKTX_190129_01_A01</td><td>AAACCCACAAAGTGTA-21L8TX_180927_001_A01</td><td>Exc L3-5 RORB LINC01202    </td><td>Glutamatergic</td><td>L5 IT  </td><td>F</td><td>M1</td><td>all</td><td>CS1912131087</td><td>Exc L3-5 RORB LINC01202    </td><td>       </td><td>Neuron 87   </td><td>H18.30.001</td><td>nucleus</td><td>NA</td><td>False</td><td>NA</td></tr>\n\t<tr><th scope=row>4</th><td>AAACCCACACTACTTT-LKTX_190129_01_A01</td><td>AAACCCACACTACTTT-21L8TX_180927_001_A01</td><td>Exc L2 LINC00507 GLRA3     </td><td>Glutamatergic</td><td>L2/3 IT</td><td>F</td><td>M1</td><td>all</td><td>CS1912131075</td><td>Exc L2 LINC00507 GLRA3     </td><td>       </td><td>Neuron 75   </td><td>H18.30.001</td><td>nucleus</td><td>NA</td><td>False</td><td>NA</td></tr>\n\t<tr><th scope=row>5</th><td>AAACCCACAGTGAGCA-LKTX_190129_01_A01</td><td>AAACCCACAGTGAGCA-21L8TX_180927_001_A01</td><td>Oligo L2-6 OPALIN FTH1P3   </td><td>Non-Neuronal </td><td>Oligo  </td><td>F</td><td>M1</td><td>all</td><td>CS1912131120</td><td>Oligo L2-6 OPALIN FTH1P3   </td><td>       </td><td>Non-neuron 3</td><td>H18.30.001</td><td>nucleus</td><td>NA</td><td>False</td><td>NA</td></tr>\n\t<tr><th scope=row>6</th><td>AAACCCAGTCACCCTT-LKTX_190129_01_A01</td><td>AAACCCAGTCACCCTT-21L8TX_180927_001_A01</td><td>Exc L5-6 FEZF2 C9orf135-AS1</td><td>Glutamatergic</td><td>L6 CT  </td><td>F</td><td>M1</td><td>all</td><td>CS1912131101</td><td>Exc L5-6 FEZF2 C9orf135-AS1</td><td>L6 CT_2</td><td>Neuron 101  </td><td>H18.30.001</td><td>nucleus</td><td>NA</td><td>False</td><td>NA</td></tr>\n</tbody>\n</table>\n","text/markdown":"\nA data.frame: 6 × 17\n\n| <!--/--> | sample_name &lt;chr&gt; | exp_component_name &lt;chr&gt; | cluster_label &lt;chr&gt; | class_label &lt;chr&gt; | subclass_label &lt;chr&gt; | donor_sex_label &lt;chr&gt; | region_label &lt;chr&gt; | cortical_layer_label &lt;chr&gt; | cell_type_accession_label &lt;chr&gt; | cell_type_alias_label &lt;chr&gt; | cell_type_alt_alias_label &lt;chr&gt; | cell_type_designation_label &lt;chr&gt; | external_donor_name_label &lt;chr&gt; | specimen_type &lt;chr&gt; | full_genotype_label &lt;lgl&gt; | outlier_call &lt;chr&gt; | outlier_type &lt;lgl&gt; |\n|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n| 1 | AAACCCAAGGATTTCC-LKTX_190129_01_A01 | AAACCCAAGGATTTCC-21L8TX_180927_001_A01 | Inh L1-2 SST CCNJL          | GABAergic     | Sst     | F | M1 | all | CS1912131050 | Inh L1-2 SST CCNJL          | <!----> | Neuron 50    | H18.30.001 | nucleus | NA | False | NA |\n| 2 | AAACCCAAGTATGGCG-LKTX_190129_01_A01 | AAACCCAAGTATGGCG-21L8TX_180927_001_A01 | Exc L5-6 FEZF2 IFNG-AS1     | Glutamatergic | L5/6 NP | F | M1 | all | CS1912131116 | Exc L5-6 FEZF2 IFNG-AS1     | <!----> | Neuron 116   | H18.30.001 | nucleus | NA | False | NA |\n| 3 | AAACCCACAAAGTGTA-LKTX_190129_01_A01 | AAACCCACAAAGTGTA-21L8TX_180927_001_A01 | Exc L3-5 RORB LINC01202     | Glutamatergic | L5 IT   | F | M1 | all | CS1912131087 | Exc L3-5 RORB LINC01202     | <!----> | Neuron 87    | H18.30.001 | nucleus | NA | False | NA |\n| 4 | AAACCCACACTACTTT-LKTX_190129_01_A01 | AAACCCACACTACTTT-21L8TX_180927_001_A01 | Exc L2 LINC00507 GLRA3      | Glutamatergic | L2/3 IT | F | M1 | all | CS1912131075 | Exc L2 LINC00507 GLRA3      | <!----> | Neuron 75    | H18.30.001 | nucleus | NA | False | NA |\n| 5 | AAACCCACAGTGAGCA-LKTX_190129_01_A01 | AAACCCACAGTGAGCA-21L8TX_180927_001_A01 | Oligo L2-6 OPALIN FTH1P3    | Non-Neuronal  | Oligo   | F | M1 | all | CS1912131120 | Oligo L2-6 OPALIN FTH1P3    | <!----> | Non-neuron 3 | H18.30.001 | nucleus | NA | False | NA |\n| 6 | AAACCCAGTCACCCTT-LKTX_190129_01_A01 | AAACCCAGTCACCCTT-21L8TX_180927_001_A01 | Exc L5-6 FEZF2 C9orf135-AS1 | Glutamatergic | L6 CT   | F | M1 | all | CS1912131101 | Exc L5-6 FEZF2 C9orf135-AS1 | L6 CT_2 | Neuron 101   | H18.30.001 | nucleus | NA | False | NA |\n\n","text/latex":"A data.frame: 6 × 17\n\\begin{tabular}{r|lllllllllllllllll}\n  & sample\\_name & exp\\_component\\_name & cluster\\_label & class\\_label & subclass\\_label & donor\\_sex\\_label & region\\_label & cortical\\_layer\\_label & cell\\_type\\_accession\\_label & cell\\_type\\_alias\\_label & cell\\_type\\_alt\\_alias\\_label & cell\\_type\\_designation\\_label & external\\_donor\\_name\\_label & specimen\\_type & full\\_genotype\\_label & outlier\\_call & outlier\\_type\\\\\n  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <lgl> & <chr> & <lgl>\\\\\n\\hline\n\t1 & AAACCCAAGGATTTCC-LKTX\\_190129\\_01\\_A01 & AAACCCAAGGATTTCC-21L8TX\\_180927\\_001\\_A01 & Inh L1-2 SST CCNJL          & GABAergic     & Sst     & F & M1 & all & CS1912131050 & Inh L1-2 SST CCNJL          &         & Neuron 50    & H18.30.001 & nucleus & NA & False & NA\\\\\n\t2 & AAACCCAAGTATGGCG-LKTX\\_190129\\_01\\_A01 & AAACCCAAGTATGGCG-21L8TX\\_180927\\_001\\_A01 & Exc L5-6 FEZF2 IFNG-AS1     & Glutamatergic & L5/6 NP & F & M1 & all & CS1912131116 & Exc L5-6 FEZF2 IFNG-AS1     &         & Neuron 116   & H18.30.001 & nucleus & NA & False & NA\\\\\n\t3 & AAACCCACAAAGTGTA-LKTX\\_190129\\_01\\_A01 & AAACCCACAAAGTGTA-21L8TX\\_180927\\_001\\_A01 & Exc L3-5 RORB LINC01202     & Glutamatergic & L5 IT   & F & M1 & all & CS1912131087 & Exc L3-5 RORB LINC01202     &         & Neuron 87    & H18.30.001 & nucleus & NA & False & NA\\\\\n\t4 & AAACCCACACTACTTT-LKTX\\_190129\\_01\\_A01 & AAACCCACACTACTTT-21L8TX\\_180927\\_001\\_A01 & Exc L2 LINC00507 GLRA3      & Glutamatergic & L2/3 IT & F & M1 & all & CS1912131075 & Exc L2 LINC00507 GLRA3      &         & Neuron 75    & H18.30.001 & nucleus & NA & False & NA\\\\\n\t5 & AAACCCACAGTGAGCA-LKTX\\_190129\\_01\\_A01 & AAACCCACAGTGAGCA-21L8TX\\_180927\\_001\\_A01 & Oligo L2-6 OPALIN FTH1P3    & Non-Neuronal  & Oligo   & F & M1 & all & CS1912131120 & Oligo L2-6 OPALIN FTH1P3    &         & Non-neuron 3 & H18.30.001 & nucleus & NA & False & NA\\\\\n\t6 & AAACCCAGTCACCCTT-LKTX\\_190129\\_01\\_A01 & AAACCCAGTCACCCTT-21L8TX\\_180927\\_001\\_A01 & Exc L5-6 FEZF2 C9orf135-AS1 & Glutamatergic & L6 CT   & F & M1 & all & CS1912131101 & Exc L5-6 FEZF2 C9orf135-AS1 & L6 CT\\_2 & Neuron 101   & H18.30.001 & nucleus & NA & False & NA\\\\\n\\end{tabular}\n","text/plain":"  sample_name                         exp_component_name                    \n1 AAACCCAAGGATTTCC-LKTX_190129_01_A01 AAACCCAAGGATTTCC-21L8TX_180927_001_A01\n2 AAACCCAAGTATGGCG-LKTX_190129_01_A01 AAACCCAAGTATGGCG-21L8TX_180927_001_A01\n3 AAACCCACAAAGTGTA-LKTX_190129_01_A01 AAACCCACAAAGTGTA-21L8TX_180927_001_A01\n4 AAACCCACACTACTTT-LKTX_190129_01_A01 AAACCCACACTACTTT-21L8TX_180927_001_A01\n5 AAACCCACAGTGAGCA-LKTX_190129_01_A01 AAACCCACAGTGAGCA-21L8TX_180927_001_A01\n6 AAACCCAGTCACCCTT-LKTX_190129_01_A01 AAACCCAGTCACCCTT-21L8TX_180927_001_A01\n  cluster_label               class_label   subclass_label donor_sex_label\n1 Inh L1-2 SST CCNJL          GABAergic     Sst            F              \n2 Exc L5-6 FEZF2 IFNG-AS1     Glutamatergic L5/6 NP        F              \n3 Exc L3-5 RORB LINC01202     Glutamatergic L5 IT          F              \n4 Exc L2 LINC00507 GLRA3      Glutamatergic L2/3 IT        F              \n5 Oligo L2-6 OPALIN FTH1P3    Non-Neuronal  Oligo          F              \n6 Exc L5-6 FEZF2 C9orf135-AS1 Glutamatergic L6 CT          F              \n  region_label cortical_layer_label cell_type_accession_label\n1 M1           all                  CS1912131050             \n2 M1           all                  CS1912131116             \n3 M1           all                  CS1912131087             \n4 M1           all                  CS1912131075             \n5 M1           all                  CS1912131120             \n6 M1           all                  CS1912131101             \n  cell_type_alias_label       cell_type_alt_alias_label\n1 Inh L1-2 SST CCNJL                                   \n2 Exc L5-6 FEZF2 IFNG-AS1                              \n3 Exc L3-5 RORB LINC01202                              \n4 Exc L2 LINC00507 GLRA3                               \n5 Oligo L2-6 OPALIN FTH1P3                             \n6 Exc L5-6 FEZF2 C9orf135-AS1 L6 CT_2                  \n  cell_type_designation_label external_donor_name_label specimen_type\n1 Neuron 50                   H18.30.001                nucleus      \n2 Neuron 116                  H18.30.001                nucleus      \n3 Neuron 87                   H18.30.001                nucleus      \n4 Neuron 75                   H18.30.001                nucleus      \n5 Non-neuron 3                H18.30.001                nucleus      \n6 Neuron 101                  H18.30.001                nucleus      \n  full_genotype_label outlier_call outlier_type\n1 NA                  False        NA          \n2 NA                  False        NA          \n3 NA                  False        NA          \n4 NA                  False        NA          \n5 NA                  False        NA          \n6 NA                  False        NA          "},"metadata":{}}],"execution_count":5},{"cell_type":"code","source":"# To show that these columns only contain NA values\nprint(sum(is.na(meta_data$outlier_type)))\nprint(sum(is.na(meta_data$full_genotype_label)))\nprint(nrow(meta_data))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-18T08:07:22.470614Z","iopub.execute_input":"2025-06-18T08:07:22.472761Z","iopub.status.idle":"2025-06-18T08:07:22.506119Z","shell.execute_reply":"2025-06-18T08:07:22.503310Z"}},"outputs":[{"name":"stdout","text":"[1] 76533\n[1] 76533\n[1] 76533\n","output_type":"stream"}],"execution_count":7},{"cell_type":"code","source":"meta_data <- select(meta_data, -full_genotype_label, -outlier_type)\nhead(meta_data)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-21T07:41:38.089892Z","iopub.execute_input":"2025-06-21T07:41:38.092013Z","iopub.status.idle":"2025-06-21T07:41:38.144419Z","shell.execute_reply":"2025-06-21T07:41:38.141934Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<table class=\"dataframe\">\n<caption>A data.frame: 6 × 15</caption>\n<thead>\n\t<tr><th></th><th scope=col>sample_name</th><th scope=col>exp_component_name</th><th scope=col>cluster_label</th><th scope=col>class_label</th><th scope=col>subclass_label</th><th scope=col>donor_sex_label</th><th scope=col>region_label</th><th scope=col>cortical_layer_label</th><th scope=col>cell_type_accession_label</th><th scope=col>cell_type_alias_label</th><th scope=col>cell_type_alt_alias_label</th><th scope=col>cell_type_designation_label</th><th scope=col>external_donor_name_label</th><th scope=col>specimen_type</th><th scope=col>outlier_call</th></tr>\n\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n</thead>\n<tbody>\n\t<tr><th scope=row>1</th><td>AAACCCAAGGATTTCC-LKTX_190129_01_A01</td><td>AAACCCAAGGATTTCC-21L8TX_180927_001_A01</td><td>Inh L1-2 SST CCNJL         </td><td>GABAergic    </td><td>Sst    </td><td>F</td><td>M1</td><td>all</td><td>CS1912131050</td><td>Inh L1-2 SST CCNJL         </td><td>       </td><td>Neuron 50   </td><td>H18.30.001</td><td>nucleus</td><td>False</td></tr>\n\t<tr><th scope=row>2</th><td>AAACCCAAGTATGGCG-LKTX_190129_01_A01</td><td>AAACCCAAGTATGGCG-21L8TX_180927_001_A01</td><td>Exc L5-6 FEZF2 IFNG-AS1    </td><td>Glutamatergic</td><td>L5/6 NP</td><td>F</td><td>M1</td><td>all</td><td>CS1912131116</td><td>Exc L5-6 FEZF2 IFNG-AS1    </td><td>       </td><td>Neuron 116  </td><td>H18.30.001</td><td>nucleus</td><td>False</td></tr>\n\t<tr><th scope=row>3</th><td>AAACCCACAAAGTGTA-LKTX_190129_01_A01</td><td>AAACCCACAAAGTGTA-21L8TX_180927_001_A01</td><td>Exc L3-5 RORB LINC01202    </td><td>Glutamatergic</td><td>L5 IT  </td><td>F</td><td>M1</td><td>all</td><td>CS1912131087</td><td>Exc L3-5 RORB LINC01202    </td><td>       </td><td>Neuron 87   </td><td>H18.30.001</td><td>nucleus</td><td>False</td></tr>\n\t<tr><th scope=row>4</th><td>AAACCCACACTACTTT-LKTX_190129_01_A01</td><td>AAACCCACACTACTTT-21L8TX_180927_001_A01</td><td>Exc L2 LINC00507 GLRA3     </td><td>Glutamatergic</td><td>L2/3 IT</td><td>F</td><td>M1</td><td>all</td><td>CS1912131075</td><td>Exc L2 LINC00507 GLRA3     </td><td>       </td><td>Neuron 75   </td><td>H18.30.001</td><td>nucleus</td><td>False</td></tr>\n\t<tr><th scope=row>5</th><td>AAACCCACAGTGAGCA-LKTX_190129_01_A01</td><td>AAACCCACAGTGAGCA-21L8TX_180927_001_A01</td><td>Oligo L2-6 OPALIN FTH1P3   </td><td>Non-Neuronal </td><td>Oligo  </td><td>F</td><td>M1</td><td>all</td><td>CS1912131120</td><td>Oligo L2-6 OPALIN FTH1P3   </td><td>       </td><td>Non-neuron 3</td><td>H18.30.001</td><td>nucleus</td><td>False</td></tr>\n\t<tr><th scope=row>6</th><td>AAACCCAGTCACCCTT-LKTX_190129_01_A01</td><td>AAACCCAGTCACCCTT-21L8TX_180927_001_A01</td><td>Exc L5-6 FEZF2 C9orf135-AS1</td><td>Glutamatergic</td><td>L6 CT  </td><td>F</td><td>M1</td><td>all</td><td>CS1912131101</td><td>Exc L5-6 FEZF2 C9orf135-AS1</td><td>L6 CT_2</td><td>Neuron 101  </td><td>H18.30.001</td><td>nucleus</td><td>False</td></tr>\n</tbody>\n</table>\n","text/markdown":"\nA data.frame: 6 × 15\n\n| <!--/--> | sample_name &lt;chr&gt; | exp_component_name &lt;chr&gt; | cluster_label &lt;chr&gt; | class_label &lt;chr&gt; | subclass_label &lt;chr&gt; | donor_sex_label &lt;chr&gt; | region_label &lt;chr&gt; | cortical_layer_label &lt;chr&gt; | cell_type_accession_label &lt;chr&gt; | cell_type_alias_label &lt;chr&gt; | cell_type_alt_alias_label &lt;chr&gt; | cell_type_designation_label &lt;chr&gt; | external_donor_name_label &lt;chr&gt; | specimen_type &lt;chr&gt; | outlier_call &lt;chr&gt; |\n|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n| 1 | AAACCCAAGGATTTCC-LKTX_190129_01_A01 | AAACCCAAGGATTTCC-21L8TX_180927_001_A01 | Inh L1-2 SST CCNJL          | GABAergic     | Sst     | F | M1 | all | CS1912131050 | Inh L1-2 SST CCNJL          | <!----> | Neuron 50    | H18.30.001 | nucleus | False |\n| 2 | AAACCCAAGTATGGCG-LKTX_190129_01_A01 | AAACCCAAGTATGGCG-21L8TX_180927_001_A01 | Exc L5-6 FEZF2 IFNG-AS1     | Glutamatergic | L5/6 NP | F | M1 | all | CS1912131116 | Exc L5-6 FEZF2 IFNG-AS1     | <!----> | Neuron 116   | H18.30.001 | nucleus | False |\n| 3 | AAACCCACAAAGTGTA-LKTX_190129_01_A01 | AAACCCACAAAGTGTA-21L8TX_180927_001_A01 | Exc L3-5 RORB LINC01202     | Glutamatergic | L5 IT   | F | M1 | all | CS1912131087 | Exc L3-5 RORB LINC01202     | <!----> | Neuron 87    | H18.30.001 | nucleus | False |\n| 4 | AAACCCACACTACTTT-LKTX_190129_01_A01 | AAACCCACACTACTTT-21L8TX_180927_001_A01 | Exc L2 LINC00507 GLRA3      | Glutamatergic | L2/3 IT | F | M1 | all | CS1912131075 | Exc L2 LINC00507 GLRA3      | <!----> | Neuron 75    | H18.30.001 | nucleus | False |\n| 5 | AAACCCACAGTGAGCA-LKTX_190129_01_A01 | AAACCCACAGTGAGCA-21L8TX_180927_001_A01 | Oligo L2-6 OPALIN FTH1P3    | Non-Neuronal  | Oligo   | F | M1 | all | CS1912131120 | Oligo L2-6 OPALIN FTH1P3    | <!----> | Non-neuron 3 | H18.30.001 | nucleus | False |\n| 6 | AAACCCAGTCACCCTT-LKTX_190129_01_A01 | AAACCCAGTCACCCTT-21L8TX_180927_001_A01 | Exc L5-6 FEZF2 C9orf135-AS1 | Glutamatergic | L6 CT   | F | M1 | all | CS1912131101 | Exc L5-6 FEZF2 C9orf135-AS1 | L6 CT_2 | Neuron 101   | H18.30.001 | nucleus | False |\n\n","text/latex":"A data.frame: 6 × 15\n\\begin{tabular}{r|lllllllllllllll}\n  & sample\\_name & exp\\_component\\_name & cluster\\_label & class\\_label & subclass\\_label & donor\\_sex\\_label & region\\_label & cortical\\_layer\\_label & cell\\_type\\_accession\\_label & cell\\_type\\_alias\\_label & cell\\_type\\_alt\\_alias\\_label & cell\\_type\\_designation\\_label & external\\_donor\\_name\\_label & specimen\\_type & outlier\\_call\\\\\n  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n\\hline\n\t1 & AAACCCAAGGATTTCC-LKTX\\_190129\\_01\\_A01 & AAACCCAAGGATTTCC-21L8TX\\_180927\\_001\\_A01 & Inh L1-2 SST CCNJL          & GABAergic     & Sst     & F & M1 & all & CS1912131050 & Inh L1-2 SST CCNJL          &         & Neuron 50    & H18.30.001 & nucleus & False\\\\\n\t2 & AAACCCAAGTATGGCG-LKTX\\_190129\\_01\\_A01 & AAACCCAAGTATGGCG-21L8TX\\_180927\\_001\\_A01 & Exc L5-6 FEZF2 IFNG-AS1     & Glutamatergic & L5/6 NP & F & M1 & all & CS1912131116 & Exc L5-6 FEZF2 IFNG-AS1     &         & Neuron 116   & H18.30.001 & nucleus & False\\\\\n\t3 & AAACCCACAAAGTGTA-LKTX\\_190129\\_01\\_A01 & AAACCCACAAAGTGTA-21L8TX\\_180927\\_001\\_A01 & Exc L3-5 RORB LINC01202     & Glutamatergic & L5 IT   & F & M1 & all & CS1912131087 & Exc L3-5 RORB LINC01202     &         & Neuron 87    & H18.30.001 & nucleus & False\\\\\n\t4 & AAACCCACACTACTTT-LKTX\\_190129\\_01\\_A01 & AAACCCACACTACTTT-21L8TX\\_180927\\_001\\_A01 & Exc L2 LINC00507 GLRA3      & Glutamatergic & L2/3 IT & F & M1 & all & CS1912131075 & Exc L2 LINC00507 GLRA3      &         & Neuron 75    & H18.30.001 & nucleus & False\\\\\n\t5 & AAACCCACAGTGAGCA-LKTX\\_190129\\_01\\_A01 & AAACCCACAGTGAGCA-21L8TX\\_180927\\_001\\_A01 & Oligo L2-6 OPALIN FTH1P3    & Non-Neuronal  & Oligo   & F & M1 & all & CS1912131120 & Oligo L2-6 OPALIN FTH1P3    &         & Non-neuron 3 & H18.30.001 & nucleus & False\\\\\n\t6 & AAACCCAGTCACCCTT-LKTX\\_190129\\_01\\_A01 & AAACCCAGTCACCCTT-21L8TX\\_180927\\_001\\_A01 & Exc L5-6 FEZF2 C9orf135-AS1 & Glutamatergic & L6 CT   & F & M1 & all & CS1912131101 & Exc L5-6 FEZF2 C9orf135-AS1 & L6 CT\\_2 & Neuron 101   & H18.30.001 & nucleus & False\\\\\n\\end{tabular}\n","text/plain":"  sample_name                         exp_component_name                    \n1 AAACCCAAGGATTTCC-LKTX_190129_01_A01 AAACCCAAGGATTTCC-21L8TX_180927_001_A01\n2 AAACCCAAGTATGGCG-LKTX_190129_01_A01 AAACCCAAGTATGGCG-21L8TX_180927_001_A01\n3 AAACCCACAAAGTGTA-LKTX_190129_01_A01 AAACCCACAAAGTGTA-21L8TX_180927_001_A01\n4 AAACCCACACTACTTT-LKTX_190129_01_A01 AAACCCACACTACTTT-21L8TX_180927_001_A01\n5 AAACCCACAGTGAGCA-LKTX_190129_01_A01 AAACCCACAGTGAGCA-21L8TX_180927_001_A01\n6 AAACCCAGTCACCCTT-LKTX_190129_01_A01 AAACCCAGTCACCCTT-21L8TX_180927_001_A01\n  cluster_label               class_label   subclass_label donor_sex_label\n1 Inh L1-2 SST CCNJL          GABAergic     Sst            F              \n2 Exc L5-6 FEZF2 IFNG-AS1     Glutamatergic L5/6 NP        F              \n3 Exc L3-5 RORB LINC01202     Glutamatergic L5 IT          F              \n4 Exc L2 LINC00507 GLRA3      Glutamatergic L2/3 IT        F              \n5 Oligo L2-6 OPALIN FTH1P3    Non-Neuronal  Oligo          F              \n6 Exc L5-6 FEZF2 C9orf135-AS1 Glutamatergic L6 CT          F              \n  region_label cortical_layer_label cell_type_accession_label\n1 M1           all                  CS1912131050             \n2 M1           all                  CS1912131116             \n3 M1           all                  CS1912131087             \n4 M1           all                  CS1912131075             \n5 M1           all                  CS1912131120             \n6 M1           all                  CS1912131101             \n  cell_type_alias_label       cell_type_alt_alias_label\n1 Inh L1-2 SST CCNJL                                   \n2 Exc L5-6 FEZF2 IFNG-AS1                              \n3 Exc L3-5 RORB LINC01202                              \n4 Exc L2 LINC00507 GLRA3                               \n5 Oligo L2-6 OPALIN FTH1P3                             \n6 Exc L5-6 FEZF2 C9orf135-AS1 L6 CT_2                  \n  cell_type_designation_label external_donor_name_label specimen_type\n1 Neuron 50                   H18.30.001                nucleus      \n2 Neuron 116                  H18.30.001                nucleus      \n3 Neuron 87                   H18.30.001                nucleus      \n4 Neuron 75                   H18.30.001                nucleus      \n5 Non-neuron 3                H18.30.001                nucleus      \n6 Neuron 101                  H18.30.001                nucleus      \n  outlier_call\n1 False       \n2 False       \n3 False       \n4 False       \n5 False       \n6 False       "},"metadata":{}}],"execution_count":8},{"cell_type":"code","source":"# Count the number of blank entries in the cell_type_alt_alias_label column\nnum_blank <- sum(meta_data$cell_type_alt_alias_label==\"\")\nprint(num_blank)\n\n# Count the number of values in this column\nprint(nrow(meta_data) - num_blank)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-18T08:16:06.952994Z","iopub.execute_input":"2025-06-18T08:16:06.955581Z","iopub.status.idle":"2025-06-18T08:16:06.989115Z","shell.execute_reply":"2025-06-18T08:16:06.986646Z"}},"outputs":[{"name":"stdout","text":"[1] 71468\n[1] 5065\n","output_type":"stream"}],"execution_count":11},{"cell_type":"code","source":"meta_data <- select(meta_data, -cell_type_alt_alias_label)\nhead(meta_data)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-21T07:41:45.739854Z","iopub.execute_input":"2025-06-21T07:41:45.741922Z","iopub.status.idle":"2025-06-21T07:41:45.844772Z","shell.execute_reply":"2025-06-21T07:41:45.842617Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<table class=\"dataframe\">\n<caption>A data.frame: 6 × 14</caption>\n<thead>\n\t<tr><th></th><th scope=col>sample_name</th><th scope=col>exp_component_name</th><th scope=col>cluster_label</th><th scope=col>class_label</th><th scope=col>subclass_label</th><th scope=col>donor_sex_label</th><th scope=col>region_label</th><th scope=col>cortical_layer_label</th><th scope=col>cell_type_accession_label</th><th scope=col>cell_type_alias_label</th><th scope=col>cell_type_designation_label</th><th scope=col>external_donor_name_label</th><th scope=col>specimen_type</th><th scope=col>outlier_call</th></tr>\n\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n</thead>\n<tbody>\n\t<tr><th scope=row>1</th><td>AAACCCAAGGATTTCC-LKTX_190129_01_A01</td><td>AAACCCAAGGATTTCC-21L8TX_180927_001_A01</td><td>Inh L1-2 SST CCNJL         </td><td>GABAergic    </td><td>Sst    </td><td>F</td><td>M1</td><td>all</td><td>CS1912131050</td><td>Inh L1-2 SST CCNJL         </td><td>Neuron 50   </td><td>H18.30.001</td><td>nucleus</td><td>False</td></tr>\n\t<tr><th scope=row>2</th><td>AAACCCAAGTATGGCG-LKTX_190129_01_A01</td><td>AAACCCAAGTATGGCG-21L8TX_180927_001_A01</td><td>Exc L5-6 FEZF2 IFNG-AS1    </td><td>Glutamatergic</td><td>L5/6 NP</td><td>F</td><td>M1</td><td>all</td><td>CS1912131116</td><td>Exc L5-6 FEZF2 IFNG-AS1    </td><td>Neuron 116  </td><td>H18.30.001</td><td>nucleus</td><td>False</td></tr>\n\t<tr><th scope=row>3</th><td>AAACCCACAAAGTGTA-LKTX_190129_01_A01</td><td>AAACCCACAAAGTGTA-21L8TX_180927_001_A01</td><td>Exc L3-5 RORB LINC01202    </td><td>Glutamatergic</td><td>L5 IT  </td><td>F</td><td>M1</td><td>all</td><td>CS1912131087</td><td>Exc L3-5 RORB LINC01202    </td><td>Neuron 87   </td><td>H18.30.001</td><td>nucleus</td><td>False</td></tr>\n\t<tr><th scope=row>4</th><td>AAACCCACACTACTTT-LKTX_190129_01_A01</td><td>AAACCCACACTACTTT-21L8TX_180927_001_A01</td><td>Exc L2 LINC00507 GLRA3     </td><td>Glutamatergic</td><td>L2/3 IT</td><td>F</td><td>M1</td><td>all</td><td>CS1912131075</td><td>Exc L2 LINC00507 GLRA3     </td><td>Neuron 75   </td><td>H18.30.001</td><td>nucleus</td><td>False</td></tr>\n\t<tr><th scope=row>5</th><td>AAACCCACAGTGAGCA-LKTX_190129_01_A01</td><td>AAACCCACAGTGAGCA-21L8TX_180927_001_A01</td><td>Oligo L2-6 OPALIN FTH1P3   </td><td>Non-Neuronal </td><td>Oligo  </td><td>F</td><td>M1</td><td>all</td><td>CS1912131120</td><td>Oligo L2-6 OPALIN FTH1P3   </td><td>Non-neuron 3</td><td>H18.30.001</td><td>nucleus</td><td>False</td></tr>\n\t<tr><th scope=row>6</th><td>AAACCCAGTCACCCTT-LKTX_190129_01_A01</td><td>AAACCCAGTCACCCTT-21L8TX_180927_001_A01</td><td>Exc L5-6 FEZF2 C9orf135-AS1</td><td>Glutamatergic</td><td>L6 CT  </td><td>F</td><td>M1</td><td>all</td><td>CS1912131101</td><td>Exc L5-6 FEZF2 C9orf135-AS1</td><td>Neuron 101  </td><td>H18.30.001</td><td>nucleus</td><td>False</td></tr>\n</tbody>\n</table>\n","text/markdown":"\nA data.frame: 6 × 14\n\n| <!--/--> | sample_name &lt;chr&gt; | exp_component_name &lt;chr&gt; | cluster_label &lt;chr&gt; | class_label &lt;chr&gt; | subclass_label &lt;chr&gt; | donor_sex_label &lt;chr&gt; | region_label &lt;chr&gt; | cortical_layer_label &lt;chr&gt; | cell_type_accession_label &lt;chr&gt; | cell_type_alias_label &lt;chr&gt; | cell_type_designation_label &lt;chr&gt; | external_donor_name_label &lt;chr&gt; | specimen_type &lt;chr&gt; | outlier_call &lt;chr&gt; |\n|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n| 1 | AAACCCAAGGATTTCC-LKTX_190129_01_A01 | AAACCCAAGGATTTCC-21L8TX_180927_001_A01 | Inh L1-2 SST CCNJL          | GABAergic     | Sst     | F | M1 | all | CS1912131050 | Inh L1-2 SST CCNJL          | Neuron 50    | H18.30.001 | nucleus | False |\n| 2 | AAACCCAAGTATGGCG-LKTX_190129_01_A01 | AAACCCAAGTATGGCG-21L8TX_180927_001_A01 | Exc L5-6 FEZF2 IFNG-AS1     | Glutamatergic | L5/6 NP | F | M1 | all | CS1912131116 | Exc L5-6 FEZF2 IFNG-AS1     | Neuron 116   | H18.30.001 | nucleus | False |\n| 3 | AAACCCACAAAGTGTA-LKTX_190129_01_A01 | AAACCCACAAAGTGTA-21L8TX_180927_001_A01 | Exc L3-5 RORB LINC01202     | Glutamatergic | L5 IT   | F | M1 | all | CS1912131087 | Exc L3-5 RORB LINC01202     | Neuron 87    | H18.30.001 | nucleus | False |\n| 4 | AAACCCACACTACTTT-LKTX_190129_01_A01 | AAACCCACACTACTTT-21L8TX_180927_001_A01 | Exc L2 LINC00507 GLRA3      | Glutamatergic | L2/3 IT | F | M1 | all | CS1912131075 | Exc L2 LINC00507 GLRA3      | Neuron 75    | H18.30.001 | nucleus | False |\n| 5 | AAACCCACAGTGAGCA-LKTX_190129_01_A01 | AAACCCACAGTGAGCA-21L8TX_180927_001_A01 | Oligo L2-6 OPALIN FTH1P3    | Non-Neuronal  | Oligo   | F | M1 | all | CS1912131120 | Oligo L2-6 OPALIN FTH1P3    | Non-neuron 3 | H18.30.001 | nucleus | False |\n| 6 | AAACCCAGTCACCCTT-LKTX_190129_01_A01 | AAACCCAGTCACCCTT-21L8TX_180927_001_A01 | Exc L5-6 FEZF2 C9orf135-AS1 | Glutamatergic | L6 CT   | F | M1 | all | CS1912131101 | Exc L5-6 FEZF2 C9orf135-AS1 | Neuron 101   | H18.30.001 | nucleus | False |\n\n","text/latex":"A data.frame: 6 × 14\n\\begin{tabular}{r|llllllllllllll}\n  & sample\\_name & exp\\_component\\_name & cluster\\_label & class\\_label & subclass\\_label & donor\\_sex\\_label & region\\_label & cortical\\_layer\\_label & cell\\_type\\_accession\\_label & cell\\_type\\_alias\\_label & cell\\_type\\_designation\\_label & external\\_donor\\_name\\_label & specimen\\_type & outlier\\_call\\\\\n  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n\\hline\n\t1 & AAACCCAAGGATTTCC-LKTX\\_190129\\_01\\_A01 & AAACCCAAGGATTTCC-21L8TX\\_180927\\_001\\_A01 & Inh L1-2 SST CCNJL          & GABAergic     & Sst     & F & M1 & all & CS1912131050 & Inh L1-2 SST CCNJL          & Neuron 50    & H18.30.001 & nucleus & False\\\\\n\t2 & AAACCCAAGTATGGCG-LKTX\\_190129\\_01\\_A01 & AAACCCAAGTATGGCG-21L8TX\\_180927\\_001\\_A01 & Exc L5-6 FEZF2 IFNG-AS1     & Glutamatergic & L5/6 NP & F & M1 & all & CS1912131116 & Exc L5-6 FEZF2 IFNG-AS1     & Neuron 116   & H18.30.001 & nucleus & False\\\\\n\t3 & AAACCCACAAAGTGTA-LKTX\\_190129\\_01\\_A01 & AAACCCACAAAGTGTA-21L8TX\\_180927\\_001\\_A01 & Exc L3-5 RORB LINC01202     & Glutamatergic & L5 IT   & F & M1 & all & CS1912131087 & Exc L3-5 RORB LINC01202     & Neuron 87    & H18.30.001 & nucleus & False\\\\\n\t4 & AAACCCACACTACTTT-LKTX\\_190129\\_01\\_A01 & AAACCCACACTACTTT-21L8TX\\_180927\\_001\\_A01 & Exc L2 LINC00507 GLRA3      & Glutamatergic & L2/3 IT & F & M1 & all & CS1912131075 & Exc L2 LINC00507 GLRA3      & Neuron 75    & H18.30.001 & nucleus & False\\\\\n\t5 & AAACCCACAGTGAGCA-LKTX\\_190129\\_01\\_A01 & AAACCCACAGTGAGCA-21L8TX\\_180927\\_001\\_A01 & Oligo L2-6 OPALIN FTH1P3    & Non-Neuronal  & Oligo   & F & M1 & all & CS1912131120 & Oligo L2-6 OPALIN FTH1P3    & Non-neuron 3 & H18.30.001 & nucleus & False\\\\\n\t6 & AAACCCAGTCACCCTT-LKTX\\_190129\\_01\\_A01 & AAACCCAGTCACCCTT-21L8TX\\_180927\\_001\\_A01 & Exc L5-6 FEZF2 C9orf135-AS1 & Glutamatergic & L6 CT   & F & M1 & all & CS1912131101 & Exc L5-6 FEZF2 C9orf135-AS1 & Neuron 101   & H18.30.001 & nucleus & False\\\\\n\\end{tabular}\n","text/plain":"  sample_name                         exp_component_name                    \n1 AAACCCAAGGATTTCC-LKTX_190129_01_A01 AAACCCAAGGATTTCC-21L8TX_180927_001_A01\n2 AAACCCAAGTATGGCG-LKTX_190129_01_A01 AAACCCAAGTATGGCG-21L8TX_180927_001_A01\n3 AAACCCACAAAGTGTA-LKTX_190129_01_A01 AAACCCACAAAGTGTA-21L8TX_180927_001_A01\n4 AAACCCACACTACTTT-LKTX_190129_01_A01 AAACCCACACTACTTT-21L8TX_180927_001_A01\n5 AAACCCACAGTGAGCA-LKTX_190129_01_A01 AAACCCACAGTGAGCA-21L8TX_180927_001_A01\n6 AAACCCAGTCACCCTT-LKTX_190129_01_A01 AAACCCAGTCACCCTT-21L8TX_180927_001_A01\n  cluster_label               class_label   subclass_label donor_sex_label\n1 Inh L1-2 SST CCNJL          GABAergic     Sst            F              \n2 Exc L5-6 FEZF2 IFNG-AS1     Glutamatergic L5/6 NP        F              \n3 Exc L3-5 RORB LINC01202     Glutamatergic L5 IT          F              \n4 Exc L2 LINC00507 GLRA3      Glutamatergic L2/3 IT        F              \n5 Oligo L2-6 OPALIN FTH1P3    Non-Neuronal  Oligo          F              \n6 Exc L5-6 FEZF2 C9orf135-AS1 Glutamatergic L6 CT          F              \n  region_label cortical_layer_label cell_type_accession_label\n1 M1           all                  CS1912131050             \n2 M1           all                  CS1912131116             \n3 M1           all                  CS1912131087             \n4 M1           all                  CS1912131075             \n5 M1           all                  CS1912131120             \n6 M1           all                  CS1912131101             \n  cell_type_alias_label       cell_type_designation_label\n1 Inh L1-2 SST CCNJL          Neuron 50                  \n2 Exc L5-6 FEZF2 IFNG-AS1     Neuron 116                 \n3 Exc L3-5 RORB LINC01202     Neuron 87                  \n4 Exc L2 LINC00507 GLRA3      Neuron 75                  \n5 Oligo L2-6 OPALIN FTH1P3    Non-neuron 3               \n6 Exc L5-6 FEZF2 C9orf135-AS1 Neuron 101                 \n  external_donor_name_label specimen_type outlier_call\n1 H18.30.001                nucleus       False       \n2 H18.30.001                nucleus       False       \n3 H18.30.001                nucleus       False       \n4 H18.30.001                nucleus       False       \n5 H18.30.001                nucleus       False       \n6 H18.30.001                nucleus       False       "},"metadata":{}}],"execution_count":9},{"cell_type":"code","source":"#To show that these two columns provide no new information\nprint(sum(meta_data$region_label!=\"M1\"))\nprint(sum(meta_data$specimen_type!=\"nucleus\"))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-18T08:23:57.022116Z","iopub.execute_input":"2025-06-18T08:23:57.024046Z","iopub.status.idle":"2025-06-18T08:23:57.045439Z","shell.execute_reply":"2025-06-18T08:23:57.043409Z"}},"outputs":[{"name":"stdout","text":"[1] 0\n[1] 0\n","output_type":"stream"}],"execution_count":19},{"cell_type":"code","source":"meta_data <- select(meta_data, -region_label, -specimen_type)\nhead(meta_data)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-21T07:41:50.679752Z","iopub.execute_input":"2025-06-21T07:41:50.681905Z","iopub.status.idle":"2025-06-21T07:41:50.727040Z","shell.execute_reply":"2025-06-21T07:41:50.724945Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<table class=\"dataframe\">\n<caption>A data.frame: 6 × 12</caption>\n<thead>\n\t<tr><th></th><th scope=col>sample_name</th><th scope=col>exp_component_name</th><th scope=col>cluster_label</th><th scope=col>class_label</th><th scope=col>subclass_label</th><th scope=col>donor_sex_label</th><th scope=col>cortical_layer_label</th><th scope=col>cell_type_accession_label</th><th scope=col>cell_type_alias_label</th><th scope=col>cell_type_designation_label</th><th scope=col>external_donor_name_label</th><th scope=col>outlier_call</th></tr>\n\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n</thead>\n<tbody>\n\t<tr><th scope=row>1</th><td>AAACCCAAGGATTTCC-LKTX_190129_01_A01</td><td>AAACCCAAGGATTTCC-21L8TX_180927_001_A01</td><td>Inh L1-2 SST CCNJL         </td><td>GABAergic    </td><td>Sst    </td><td>F</td><td>all</td><td>CS1912131050</td><td>Inh L1-2 SST CCNJL         </td><td>Neuron 50   </td><td>H18.30.001</td><td>False</td></tr>\n\t<tr><th scope=row>2</th><td>AAACCCAAGTATGGCG-LKTX_190129_01_A01</td><td>AAACCCAAGTATGGCG-21L8TX_180927_001_A01</td><td>Exc L5-6 FEZF2 IFNG-AS1    </td><td>Glutamatergic</td><td>L5/6 NP</td><td>F</td><td>all</td><td>CS1912131116</td><td>Exc L5-6 FEZF2 IFNG-AS1    </td><td>Neuron 116  </td><td>H18.30.001</td><td>False</td></tr>\n\t<tr><th scope=row>3</th><td>AAACCCACAAAGTGTA-LKTX_190129_01_A01</td><td>AAACCCACAAAGTGTA-21L8TX_180927_001_A01</td><td>Exc L3-5 RORB LINC01202    </td><td>Glutamatergic</td><td>L5 IT  </td><td>F</td><td>all</td><td>CS1912131087</td><td>Exc L3-5 RORB LINC01202    </td><td>Neuron 87   </td><td>H18.30.001</td><td>False</td></tr>\n\t<tr><th scope=row>4</th><td>AAACCCACACTACTTT-LKTX_190129_01_A01</td><td>AAACCCACACTACTTT-21L8TX_180927_001_A01</td><td>Exc L2 LINC00507 GLRA3     </td><td>Glutamatergic</td><td>L2/3 IT</td><td>F</td><td>all</td><td>CS1912131075</td><td>Exc L2 LINC00507 GLRA3     </td><td>Neuron 75   </td><td>H18.30.001</td><td>False</td></tr>\n\t<tr><th scope=row>5</th><td>AAACCCACAGTGAGCA-LKTX_190129_01_A01</td><td>AAACCCACAGTGAGCA-21L8TX_180927_001_A01</td><td>Oligo L2-6 OPALIN FTH1P3   </td><td>Non-Neuronal </td><td>Oligo  </td><td>F</td><td>all</td><td>CS1912131120</td><td>Oligo L2-6 OPALIN FTH1P3   </td><td>Non-neuron 3</td><td>H18.30.001</td><td>False</td></tr>\n\t<tr><th scope=row>6</th><td>AAACCCAGTCACCCTT-LKTX_190129_01_A01</td><td>AAACCCAGTCACCCTT-21L8TX_180927_001_A01</td><td>Exc L5-6 FEZF2 C9orf135-AS1</td><td>Glutamatergic</td><td>L6 CT  </td><td>F</td><td>all</td><td>CS1912131101</td><td>Exc L5-6 FEZF2 C9orf135-AS1</td><td>Neuron 101  </td><td>H18.30.001</td><td>False</td></tr>\n</tbody>\n</table>\n","text/markdown":"\nA data.frame: 6 × 12\n\n| <!--/--> | sample_name &lt;chr&gt; | exp_component_name &lt;chr&gt; | cluster_label &lt;chr&gt; | class_label &lt;chr&gt; | subclass_label &lt;chr&gt; | donor_sex_label &lt;chr&gt; | cortical_layer_label &lt;chr&gt; | cell_type_accession_label &lt;chr&gt; | cell_type_alias_label &lt;chr&gt; | cell_type_designation_label &lt;chr&gt; | external_donor_name_label &lt;chr&gt; | outlier_call &lt;chr&gt; |\n|---|---|---|---|---|---|---|---|---|---|---|---|---|\n| 1 | AAACCCAAGGATTTCC-LKTX_190129_01_A01 | AAACCCAAGGATTTCC-21L8TX_180927_001_A01 | Inh L1-2 SST CCNJL          | GABAergic     | Sst     | F | all | CS1912131050 | Inh L1-2 SST CCNJL          | Neuron 50    | H18.30.001 | False |\n| 2 | AAACCCAAGTATGGCG-LKTX_190129_01_A01 | AAACCCAAGTATGGCG-21L8TX_180927_001_A01 | Exc L5-6 FEZF2 IFNG-AS1     | Glutamatergic | L5/6 NP | F | all | CS1912131116 | Exc L5-6 FEZF2 IFNG-AS1     | Neuron 116   | H18.30.001 | False |\n| 3 | AAACCCACAAAGTGTA-LKTX_190129_01_A01 | AAACCCACAAAGTGTA-21L8TX_180927_001_A01 | Exc L3-5 RORB LINC01202     | Glutamatergic | L5 IT   | F | all | CS1912131087 | Exc L3-5 RORB LINC01202     | Neuron 87    | H18.30.001 | False |\n| 4 | AAACCCACACTACTTT-LKTX_190129_01_A01 | AAACCCACACTACTTT-21L8TX_180927_001_A01 | Exc L2 LINC00507 GLRA3      | Glutamatergic | L2/3 IT | F | all | CS1912131075 | Exc L2 LINC00507 GLRA3      | Neuron 75    | H18.30.001 | False |\n| 5 | AAACCCACAGTGAGCA-LKTX_190129_01_A01 | AAACCCACAGTGAGCA-21L8TX_180927_001_A01 | Oligo L2-6 OPALIN FTH1P3    | Non-Neuronal  | Oligo   | F | all | CS1912131120 | Oligo L2-6 OPALIN FTH1P3    | Non-neuron 3 | H18.30.001 | False |\n| 6 | AAACCCAGTCACCCTT-LKTX_190129_01_A01 | AAACCCAGTCACCCTT-21L8TX_180927_001_A01 | Exc L5-6 FEZF2 C9orf135-AS1 | Glutamatergic | L6 CT   | F | all | CS1912131101 | Exc L5-6 FEZF2 C9orf135-AS1 | Neuron 101   | H18.30.001 | False |\n\n","text/latex":"A data.frame: 6 × 12\n\\begin{tabular}{r|llllllllllll}\n  & sample\\_name & exp\\_component\\_name & cluster\\_label & class\\_label & subclass\\_label & donor\\_sex\\_label & cortical\\_layer\\_label & cell\\_type\\_accession\\_label & cell\\_type\\_alias\\_label & cell\\_type\\_designation\\_label & external\\_donor\\_name\\_label & outlier\\_call\\\\\n  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n\\hline\n\t1 & AAACCCAAGGATTTCC-LKTX\\_190129\\_01\\_A01 & AAACCCAAGGATTTCC-21L8TX\\_180927\\_001\\_A01 & Inh L1-2 SST CCNJL          & GABAergic     & Sst     & F & all & CS1912131050 & Inh L1-2 SST CCNJL          & Neuron 50    & H18.30.001 & False\\\\\n\t2 & AAACCCAAGTATGGCG-LKTX\\_190129\\_01\\_A01 & AAACCCAAGTATGGCG-21L8TX\\_180927\\_001\\_A01 & Exc L5-6 FEZF2 IFNG-AS1     & Glutamatergic & L5/6 NP & F & all & CS1912131116 & Exc L5-6 FEZF2 IFNG-AS1     & Neuron 116   & H18.30.001 & False\\\\\n\t3 & AAACCCACAAAGTGTA-LKTX\\_190129\\_01\\_A01 & AAACCCACAAAGTGTA-21L8TX\\_180927\\_001\\_A01 & Exc L3-5 RORB LINC01202     & Glutamatergic & L5 IT   & F & all & CS1912131087 & Exc L3-5 RORB LINC01202     & Neuron 87    & H18.30.001 & False\\\\\n\t4 & AAACCCACACTACTTT-LKTX\\_190129\\_01\\_A01 & AAACCCACACTACTTT-21L8TX\\_180927\\_001\\_A01 & Exc L2 LINC00507 GLRA3      & Glutamatergic & L2/3 IT & F & all & CS1912131075 & Exc L2 LINC00507 GLRA3      & Neuron 75    & H18.30.001 & False\\\\\n\t5 & AAACCCACAGTGAGCA-LKTX\\_190129\\_01\\_A01 & AAACCCACAGTGAGCA-21L8TX\\_180927\\_001\\_A01 & Oligo L2-6 OPALIN FTH1P3    & Non-Neuronal  & Oligo   & F & all & CS1912131120 & Oligo L2-6 OPALIN FTH1P3    & Non-neuron 3 & H18.30.001 & False\\\\\n\t6 & AAACCCAGTCACCCTT-LKTX\\_190129\\_01\\_A01 & AAACCCAGTCACCCTT-21L8TX\\_180927\\_001\\_A01 & Exc L5-6 FEZF2 C9orf135-AS1 & Glutamatergic & L6 CT   & F & all & CS1912131101 & Exc L5-6 FEZF2 C9orf135-AS1 & Neuron 101   & H18.30.001 & False\\\\\n\\end{tabular}\n","text/plain":"  sample_name                         exp_component_name                    \n1 AAACCCAAGGATTTCC-LKTX_190129_01_A01 AAACCCAAGGATTTCC-21L8TX_180927_001_A01\n2 AAACCCAAGTATGGCG-LKTX_190129_01_A01 AAACCCAAGTATGGCG-21L8TX_180927_001_A01\n3 AAACCCACAAAGTGTA-LKTX_190129_01_A01 AAACCCACAAAGTGTA-21L8TX_180927_001_A01\n4 AAACCCACACTACTTT-LKTX_190129_01_A01 AAACCCACACTACTTT-21L8TX_180927_001_A01\n5 AAACCCACAGTGAGCA-LKTX_190129_01_A01 AAACCCACAGTGAGCA-21L8TX_180927_001_A01\n6 AAACCCAGTCACCCTT-LKTX_190129_01_A01 AAACCCAGTCACCCTT-21L8TX_180927_001_A01\n  cluster_label               class_label   subclass_label donor_sex_label\n1 Inh L1-2 SST CCNJL          GABAergic     Sst            F              \n2 Exc L5-6 FEZF2 IFNG-AS1     Glutamatergic L5/6 NP        F              \n3 Exc L3-5 RORB LINC01202     Glutamatergic L5 IT          F              \n4 Exc L2 LINC00507 GLRA3      Glutamatergic L2/3 IT        F              \n5 Oligo L2-6 OPALIN FTH1P3    Non-Neuronal  Oligo          F              \n6 Exc L5-6 FEZF2 C9orf135-AS1 Glutamatergic L6 CT          F              \n  cortical_layer_label cell_type_accession_label cell_type_alias_label      \n1 all                  CS1912131050              Inh L1-2 SST CCNJL         \n2 all                  CS1912131116              Exc L5-6 FEZF2 IFNG-AS1    \n3 all                  CS1912131087              Exc L3-5 RORB LINC01202    \n4 all                  CS1912131075              Exc L2 LINC00507 GLRA3     \n5 all                  CS1912131120              Oligo L2-6 OPALIN FTH1P3   \n6 all                  CS1912131101              Exc L5-6 FEZF2 C9orf135-AS1\n  cell_type_designation_label external_donor_name_label outlier_call\n1 Neuron 50                   H18.30.001                False       \n2 Neuron 116                  H18.30.001                False       \n3 Neuron 87                   H18.30.001                False       \n4 Neuron 75                   H18.30.001                False       \n5 Non-neuron 3                H18.30.001                False       \n6 Neuron 101                  H18.30.001                False       "},"metadata":{}}],"execution_count":10},{"cell_type":"code","source":"# To see the number of cell types, and the number of cells of each type\nselect(meta_data, class_label) %>%\n    table()","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-21T07:47:53.332885Z","iopub.execute_input":"2025-06-21T07:47:53.335463Z","iopub.status.idle":"2025-06-21T07:47:53.386188Z","shell.execute_reply":"2025-06-21T07:47:53.383503Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"class_label\n    GABAergic Glutamatergic  Non-Neuronal \n        23992         48536          4005 "},"metadata":{}}],"execution_count":11},{"cell_type":"markdown","source":"## 4. Explore and clean the counts matrix.","metadata":{}},{"cell_type":"code","source":"str(gene_matrix)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-17T11:45:00.878428Z","iopub.execute_input":"2025-06-17T11:45:00.880813Z","iopub.status.idle":"2025-06-17T11:45:01.236347Z","shell.execute_reply":"2025-06-17T11:45:01.233585Z"}},"outputs":[{"name":"stdout","text":"Classes ‘data.table’ and 'data.frame':\t76533 obs. of  50282 variables:\n $ sample_name           : chr  \"AAACCCAAGGATTTCC-LKTX_190129_01_A01\" \"AAACCCAAGTATGGCG-LKTX_190129_01_A01\" \"AAACCCACAAAGTGTA-LKTX_190129_01_A01\" \"AAACCCACACTACTTT-LKTX_190129_01_A01\" ...\n $ DDX11L1               : int  0 0 0 0 0 0 0 0 0 0 ...\n $ WASH7P                : int  0 0 0 0 0 0 0 0 0 0 ...\n $ MIR6859-1             : int  0 0 0 0 0 0 0 0 0 0 ...\n $ MIR1302-2             : int  0 0 0 0 0 0 0 0 0 0 ...\n $ FAM138A               : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC105379212          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ OR4G4P                : int  0 0 0 0 0 0 0 0 0 0 ...\n $ OR4G11P               : int  0 0 0 0 0 0 0 0 0 0 ...\n $ OR4F5                 : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC105379213          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ CICP27                : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC729737             : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC100996442          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC105379214          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC102725121          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC102723897          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ MIR6859-2             : int  0 0 0 0 0 0 0 0 0 0 ...\n $ RPL23AP21             : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC102723917          : int  0 0 0 0 0 1 0 0 0 1 ...\n $ LOC105379431          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ RPL23AP24             : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC105379248          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ OR4F29                : int  0 0 0 0 0 0 0 0 0 0 ...\n $ CICP7                 : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC100132287          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC100134822          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC105378947          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC101928626          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ MTND1P23              : int  0 0 0 0 0 0 0 0 0 0 ...\n $ MTND2P28              : int  0 0 0 0 0 0 0 0 0 0 ...\n $ MIR6723               : int  0 0 0 0 0 0 0 0 0 0 ...\n $ OR4F16                : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC105378582          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ CICP3                 : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC100133331          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC100288069          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC105378581          : int  0 0 0 0 0 0 0 0 1 0 ...\n $ LOC100287934          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC105378580          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ FAM87B                : int  0 0 0 0 0 1 0 0 0 0 ...\n $ LINC00115             : int  0 0 0 0 0 0 0 0 1 0 ...\n $ LINC01128             : int  1 0 2 0 1 0 0 1 0 0 ...\n $ FAM41C                : int  0 0 0 0 0 0 0 0 0 0 ...\n $ TUBB8P11              : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC284600             : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC100130417          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ SAMD11                : int  0 0 0 0 0 0 0 0 0 0 ...\n $ NOC2L                 : int  0 2 0 0 0 1 1 0 1 3 ...\n $ KLHL17                : int  0 0 0 0 0 0 0 0 0 1 ...\n $ PLEKHN1               : int  0 0 0 0 0 0 0 0 0 0 ...\n $ PERM1                 : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC105378583          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ HES4                  : int  0 0 1 0 0 0 1 0 2 0 ...\n $ RPL39P12              : int  0 0 0 0 0 0 0 0 0 0 ...\n $ ISG15                 : int  0 0 0 0 0 0 0 0 0 0 ...\n $ AGRN                  : int  1 0 0 0 0 0 0 0 0 2 ...\n $ LOC100288175          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC105369174          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC105378948          : int  0 0 1 0 0 0 0 0 0 1 ...\n $ RNF223                : int  0 0 0 0 0 0 0 0 0 0 ...\n $ C1orf159              : int  1 0 0 0 1 0 3 2 0 0 ...\n $ LOC105378584          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LINC01342             : int  0 0 0 0 0 0 0 0 0 0 ...\n $ MIR200B               : int  0 0 0 0 0 0 0 0 0 0 ...\n $ MIR200A               : int  0 0 0 0 0 0 0 0 0 0 ...\n $ MIR429                : int  0 0 0 0 0 0 0 0 0 0 ...\n $ TTLL10-AS1            : int  0 0 0 0 0 0 0 0 0 0 ...\n $ TTLL10                : int  0 0 0 0 0 0 0 0 0 0 ...\n $ TNFRSF18              : int  0 0 0 0 0 0 0 0 0 0 ...\n $ TNFRSF4               : int  0 0 0 0 0 0 0 0 0 0 ...\n $ SDF4                  : int  0 0 0 0 0 0 0 0 1 1 ...\n $ B3GALT6               : int  0 0 0 0 0 0 0 1 1 0 ...\n $ FAM132A               : int  0 0 0 0 0 0 0 0 0 0 ...\n $ UBE2J2                : int  0 1 1 0 0 0 0 0 0 3 ...\n $ LOC101928895          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ SCNN1D                : int  1 0 0 0 0 0 1 1 0 0 ...\n $ ACAP3                 : int  0 1 0 0 0 0 2 1 0 1 ...\n $ MIR6726               : int  0 0 0 0 0 0 0 0 0 0 ...\n $ PUSL1                 : int  0 0 0 0 0 0 0 0 0 0 ...\n $ CPSF3L                : int  0 2 0 0 0 1 0 2 0 0 ...\n $ MIR6727               : int  0 0 0 0 0 0 0 0 0 0 ...\n $ CPTP                  : int  0 0 0 0 0 0 0 0 0 0 ...\n $ TAS1R3                : int  0 0 0 0 0 0 0 0 0 0 ...\n $ DVL1                  : int  1 0 0 0 0 2 1 0 1 0 ...\n $ MIR6808               : int  0 0 0 0 0 0 0 0 0 0 ...\n $ MXRA8                 : int  0 0 0 0 0 0 0 0 0 0 ...\n $ AURKAIP1              : int  0 1 0 0 0 0 0 0 0 2 ...\n $ CCNL2                 : int  3 1 2 0 0 4 2 0 0 0 ...\n $ LOC148413             : int  0 0 0 0 0 0 0 0 0 0 ...\n $ MRPL20                : int  0 0 0 0 0 0 0 0 0 1 ...\n $ ANKRD65               : int  0 0 0 0 0 0 0 2 0 0 ...\n $ LOC105378585          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ TMEM88B               : int  0 0 0 0 0 0 0 0 0 0 ...\n $ LOC102724312          : int  0 0 0 0 0 0 0 0 0 0 ...\n $ VWA1                  : int  0 0 0 0 0 0 0 0 0 0 ...\n $ ATAD3C                : int  0 0 0 0 0 0 0 0 0 0 ...\n $ ATAD3B                : int  0 0 0 0 1 0 0 0 0 1 ...\n $ ATAD3A                : int  0 0 0 0 0 2 1 0 0 1 ...\n  [list output truncated]\n - attr(*, \".internal.selfref\")=<externalptr> \n","output_type":"stream"}],"execution_count":24},{"cell_type":"code","source":"head(gene_matrix)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-17T12:15:40.269356Z","iopub.execute_input":"2025-06-17T12:15:40.273625Z","iopub.status.idle":"2025-06-17T12:16:02.619137Z","shell.execute_reply":"2025-06-17T12:16:02.616424Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<table class=\"dataframe\">\n<caption>A data.table: 6 × 50282</caption>\n<thead>\n\t<tr><th scope=col>sample_name</th><th scope=col>DDX11L1</th><th scope=col>WASH7P</th><th scope=col>MIR6859-1</th><th scope=col>MIR1302-2</th><th scope=col>FAM138A</th><th scope=col>LOC105379212</th><th scope=col>OR4G4P</th><th scope=col>OR4G11P</th><th scope=col>OR4F5</th><th scope=col>⋯</th><th scope=col>ND4</th><th scope=col>TRNH</th><th scope=col>TRNS2</th><th scope=col>TRNL2</th><th scope=col>ND5</th><th scope=col>ND6</th><th scope=col>TRNE</th><th scope=col>CYTB</th><th scope=col>TRNT</th><th scope=col>TRNP</th></tr>\n\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n</thead>\n<tbody>\n\t<tr><td>AAACCCAAGGATTTCC-LKTX_190129_01_A01</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 8</td><td>0</td><td>0</td><td>0</td><td>2</td><td>0</td><td>0</td><td>19</td><td>0</td><td>0</td></tr>\n\t<tr><td>AAACCCAAGTATGGCG-LKTX_190129_01_A01</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 5</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td> 3</td><td>0</td><td>0</td></tr>\n\t<tr><td>AAACCCACAAAGTGTA-LKTX_190129_01_A01</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 8</td><td>0</td><td>0</td><td>0</td><td>1</td><td>0</td><td>0</td><td> 5</td><td>0</td><td>0</td></tr>\n\t<tr><td>AAACCCACACTACTTT-LKTX_190129_01_A01</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 6</td><td>0</td><td>0</td><td>0</td><td>6</td><td>0</td><td>0</td><td> 8</td><td>0</td><td>0</td></tr>\n\t<tr><td>AAACCCACAGTGAGCA-LKTX_190129_01_A01</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td> 0</td><td>0</td><td>0</td></tr>\n\t<tr><td>AAACCCAGTCACCCTT-LKTX_190129_01_A01</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>19</td><td>0</td><td>0</td><td>0</td><td>2</td><td>0</td><td>0</td><td> 7</td><td>0</td><td>0</td></tr>\n</tbody>\n</table>\n","text/markdown":"\nA data.table: 6 × 50282\n\n| sample_name &lt;chr&gt; | DDX11L1 &lt;int&gt; | WASH7P &lt;int&gt; | MIR6859-1 &lt;int&gt; | MIR1302-2 &lt;int&gt; | FAM138A &lt;int&gt; | LOC105379212 &lt;int&gt; | OR4G4P &lt;int&gt; | OR4G11P &lt;int&gt; | OR4F5 &lt;int&gt; | ⋯ ⋯ | ND4 &lt;int&gt; | TRNH &lt;int&gt; | TRNS2 &lt;int&gt; | TRNL2 &lt;int&gt; | ND5 &lt;int&gt; | ND6 &lt;int&gt; | TRNE &lt;int&gt; | CYTB &lt;int&gt; | TRNT &lt;int&gt; | TRNP &lt;int&gt; |\n|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n| AAACCCAAGGATTTCC-LKTX_190129_01_A01 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  8 | 0 | 0 | 0 | 2 | 0 | 0 | 19 | 0 | 0 |\n| AAACCCAAGTATGGCG-LKTX_190129_01_A01 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  5 | 0 | 0 | 0 | 0 | 0 | 0 |  3 | 0 | 0 |\n| AAACCCACAAAGTGTA-LKTX_190129_01_A01 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  8 | 0 | 0 | 0 | 1 | 0 | 0 |  5 | 0 | 0 |\n| AAACCCACACTACTTT-LKTX_190129_01_A01 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  6 | 0 | 0 | 0 | 6 | 0 | 0 |  8 | 0 | 0 |\n| AAACCCACAGTGAGCA-LKTX_190129_01_A01 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  1 | 0 | 0 | 0 | 0 | 0 | 0 |  0 | 0 | 0 |\n| AAACCCAGTCACCCTT-LKTX_190129_01_A01 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | 19 | 0 | 0 | 0 | 2 | 0 | 0 |  7 | 0 | 0 |\n\n","text/latex":"A data.table: 6 × 50282\n\\begin{tabular}{lllllllllllllllllllll}\n sample\\_name & DDX11L1 & WASH7P & MIR6859-1 & MIR1302-2 & FAM138A & LOC105379212 & OR4G4P & OR4G11P & OR4F5 & ⋯ & ND4 & TRNH & TRNS2 & TRNL2 & ND5 & ND6 & TRNE & CYTB & TRNT & TRNP\\\\\n <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n\\hline\n\t AAACCCAAGGATTTCC-LKTX\\_190129\\_01\\_A01 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  8 & 0 & 0 & 0 & 2 & 0 & 0 & 19 & 0 & 0\\\\\n\t AAACCCAAGTATGGCG-LKTX\\_190129\\_01\\_A01 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  5 & 0 & 0 & 0 & 0 & 0 & 0 &  3 & 0 & 0\\\\\n\t AAACCCACAAAGTGTA-LKTX\\_190129\\_01\\_A01 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  8 & 0 & 0 & 0 & 1 & 0 & 0 &  5 & 0 & 0\\\\\n\t AAACCCACACTACTTT-LKTX\\_190129\\_01\\_A01 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  6 & 0 & 0 & 0 & 6 & 0 & 0 &  8 & 0 & 0\\\\\n\t AAACCCACAGTGAGCA-LKTX\\_190129\\_01\\_A01 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  1 & 0 & 0 & 0 & 0 & 0 & 0 &  0 & 0 & 0\\\\\n\t AAACCCAGTCACCCTT-LKTX\\_190129\\_01\\_A01 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & 19 & 0 & 0 & 0 & 2 & 0 & 0 &  7 & 0 & 0\\\\\n\\end{tabular}\n","text/plain":"  sample_name                         DDX11L1 WASH7P MIR6859-1 MIR1302-2\n1 AAACCCAAGGATTTCC-LKTX_190129_01_A01 0       0      0         0        \n2 AAACCCAAGTATGGCG-LKTX_190129_01_A01 0       0      0         0        \n3 AAACCCACAAAGTGTA-LKTX_190129_01_A01 0       0      0         0        \n4 AAACCCACACTACTTT-LKTX_190129_01_A01 0       0      0         0        \n5 AAACCCACAGTGAGCA-LKTX_190129_01_A01 0       0      0         0        \n6 AAACCCAGTCACCCTT-LKTX_190129_01_A01 0       0      0         0        \n  FAM138A LOC105379212 OR4G4P OR4G11P OR4F5 ⋯ ND4 TRNH TRNS2 TRNL2 ND5 ND6 TRNE\n1 0       0            0      0       0     ⋯  8  0    0     0     2   0   0   \n2 0       0            0      0       0     ⋯  5  0    0     0     0   0   0   \n3 0       0            0      0       0     ⋯  8  0    0     0     1   0   0   \n4 0       0            0      0       0     ⋯  6  0    0     0     6   0   0   \n5 0       0            0      0       0     ⋯  1  0    0     0     0   0   0   \n6 0       0            0      0       0     ⋯ 19  0    0     0     2   0   0   \n  CYTB TRNT TRNP\n1 19   0    0   \n2  3   0    0   \n3  5   0    0   \n4  8   0    0   \n5  0   0    0   \n6  7   0    0   "},"metadata":{}}],"execution_count":28},{"cell_type":"code","source":"print(sum(duplicated(colnames(gene_matrix))))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-21T07:51:43.776975Z","iopub.execute_input":"2025-06-21T07:51:43.779390Z","iopub.status.idle":"2025-06-21T07:51:43.808112Z","shell.execute_reply":"2025-06-21T07:51:43.805467Z"}},"outputs":[{"name":"stdout","text":"[1] 0\n","output_type":"stream"}],"execution_count":12},{"cell_type":"markdown","source":"## 5. Perform Exploratory Data Analysis (EDA).","metadata":{}},{"cell_type":"code","source":"meta_subset <- meta_data %>% \n    select(sample_name, cluster_label:subclass_label)\nhead(meta_subset)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-21T07:58:13.482123Z","iopub.execute_input":"2025-06-21T07:58:13.485230Z","iopub.status.idle":"2025-06-21T07:58:13.543378Z","shell.execute_reply":"2025-06-21T07:58:13.540725Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<table class=\"dataframe\">\n<caption>A data.frame: 6 × 4</caption>\n<thead>\n\t<tr><th></th><th scope=col>sample_name</th><th scope=col>cluster_label</th><th scope=col>class_label</th><th scope=col>subclass_label</th></tr>\n\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n</thead>\n<tbody>\n\t<tr><th scope=row>1</th><td>AAACCCAAGGATTTCC-LKTX_190129_01_A01</td><td>Inh L1-2 SST CCNJL         </td><td>GABAergic    </td><td>Sst    </td></tr>\n\t<tr><th scope=row>2</th><td>AAACCCAAGTATGGCG-LKTX_190129_01_A01</td><td>Exc L5-6 FEZF2 IFNG-AS1    </td><td>Glutamatergic</td><td>L5/6 NP</td></tr>\n\t<tr><th scope=row>3</th><td>AAACCCACAAAGTGTA-LKTX_190129_01_A01</td><td>Exc L3-5 RORB LINC01202    </td><td>Glutamatergic</td><td>L5 IT  </td></tr>\n\t<tr><th scope=row>4</th><td>AAACCCACACTACTTT-LKTX_190129_01_A01</td><td>Exc L2 LINC00507 GLRA3     </td><td>Glutamatergic</td><td>L2/3 IT</td></tr>\n\t<tr><th scope=row>5</th><td>AAACCCACAGTGAGCA-LKTX_190129_01_A01</td><td>Oligo L2-6 OPALIN FTH1P3   </td><td>Non-Neuronal </td><td>Oligo  </td></tr>\n\t<tr><th scope=row>6</th><td>AAACCCAGTCACCCTT-LKTX_190129_01_A01</td><td>Exc L5-6 FEZF2 C9orf135-AS1</td><td>Glutamatergic</td><td>L6 CT  </td></tr>\n</tbody>\n</table>\n","text/markdown":"\nA data.frame: 6 × 4\n\n| <!--/--> | sample_name &lt;chr&gt; | cluster_label &lt;chr&gt; | class_label &lt;chr&gt; | subclass_label &lt;chr&gt; |\n|---|---|---|---|---|\n| 1 | AAACCCAAGGATTTCC-LKTX_190129_01_A01 | Inh L1-2 SST CCNJL          | GABAergic     | Sst     |\n| 2 | AAACCCAAGTATGGCG-LKTX_190129_01_A01 | Exc L5-6 FEZF2 IFNG-AS1     | Glutamatergic | L5/6 NP |\n| 3 | AAACCCACAAAGTGTA-LKTX_190129_01_A01 | Exc L3-5 RORB LINC01202     | Glutamatergic | L5 IT   |\n| 4 | AAACCCACACTACTTT-LKTX_190129_01_A01 | Exc L2 LINC00507 GLRA3      | Glutamatergic | L2/3 IT |\n| 5 | AAACCCACAGTGAGCA-LKTX_190129_01_A01 | Oligo L2-6 OPALIN FTH1P3    | Non-Neuronal  | Oligo   |\n| 6 | AAACCCAGTCACCCTT-LKTX_190129_01_A01 | Exc L5-6 FEZF2 C9orf135-AS1 | Glutamatergic | L6 CT   |\n\n","text/latex":"A data.frame: 6 × 4\n\\begin{tabular}{r|llll}\n  & sample\\_name & cluster\\_label & class\\_label & subclass\\_label\\\\\n  & <chr> & <chr> & <chr> & <chr>\\\\\n\\hline\n\t1 & AAACCCAAGGATTTCC-LKTX\\_190129\\_01\\_A01 & Inh L1-2 SST CCNJL          & GABAergic     & Sst    \\\\\n\t2 & AAACCCAAGTATGGCG-LKTX\\_190129\\_01\\_A01 & Exc L5-6 FEZF2 IFNG-AS1     & Glutamatergic & L5/6 NP\\\\\n\t3 & AAACCCACAAAGTGTA-LKTX\\_190129\\_01\\_A01 & Exc L3-5 RORB LINC01202     & Glutamatergic & L5 IT  \\\\\n\t4 & AAACCCACACTACTTT-LKTX\\_190129\\_01\\_A01 & Exc L2 LINC00507 GLRA3      & Glutamatergic & L2/3 IT\\\\\n\t5 & AAACCCACAGTGAGCA-LKTX\\_190129\\_01\\_A01 & Oligo L2-6 OPALIN FTH1P3    & Non-Neuronal  & Oligo  \\\\\n\t6 & AAACCCAGTCACCCTT-LKTX\\_190129\\_01\\_A01 & Exc L5-6 FEZF2 C9orf135-AS1 & Glutamatergic & L6 CT  \\\\\n\\end{tabular}\n","text/plain":"  sample_name                         cluster_label               class_label  \n1 AAACCCAAGGATTTCC-LKTX_190129_01_A01 Inh L1-2 SST CCNJL          GABAergic    \n2 AAACCCAAGTATGGCG-LKTX_190129_01_A01 Exc L5-6 FEZF2 IFNG-AS1     Glutamatergic\n3 AAACCCACAAAGTGTA-LKTX_190129_01_A01 Exc L3-5 RORB LINC01202     Glutamatergic\n4 AAACCCACACTACTTT-LKTX_190129_01_A01 Exc L2 LINC00507 GLRA3      Glutamatergic\n5 AAACCCACAGTGAGCA-LKTX_190129_01_A01 Oligo L2-6 OPALIN FTH1P3    Non-Neuronal \n6 AAACCCAGTCACCCTT-LKTX_190129_01_A01 Exc L5-6 FEZF2 C9orf135-AS1 Glutamatergic\n  subclass_label\n1 Sst           \n2 L5/6 NP       \n3 L5 IT         \n4 L2/3 IT       \n5 Oligo         \n6 L6 CT         "},"metadata":{}}],"execution_count":13}]}